<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.8"/>
<title>SnowTools: GenomicRegion Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">SnowTools
   &#160;<span id="projectnumber">0.0</span>
   </div>
   <div id="projectbrief">C++ htslib/bwa-mem interface and command line tools for interrogating sequence data</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.8 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="hierarchy.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceSnowTools.html">SnowTools</a></li><li class="navelem"><a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-static-methods">Static Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a> &#124;
<a href="#friends">Friends</a> &#124;
<a href="classSnowTools_1_1GenomicRegion-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">GenomicRegion Class Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>Container for an interval on the genome.  
 <a href="classSnowTools_1_1GenomicRegion.html#details">More...</a></p>

<p><code>#include &lt;<a class="el" href="GenomicRegion_8h_source.html">GenomicRegion.h</a>&gt;</code></p>
<div class="dynheader">
Inheritance diagram for GenomicRegion:</div>
<div class="dyncontent">
 <div class="center">
  <img src="classSnowTools_1_1GenomicRegion.png" usemap="#GenomicRegion_map" alt=""/>
  <map id="GenomicRegion_map" name="GenomicRegion_map">
<area href="classSnowTools_1_1FracRegion.html" title="Extension of GenomicRegion with a fraction of reads to keep on an interval. " alt="FracRegion" shape="rect" coords="0,56,100,80"/>
</map>
 </div></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:aac6129647c2ad07c5ee63edb3f65c705"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aac6129647c2ad07c5ee63edb3f65c705"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#aac6129647c2ad07c5ee63edb3f65c705">GenomicRegion</a> ()</td></tr>
<tr class="memdesc:aac6129647c2ad07c5ee63edb3f65c705"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct an "empty" <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> at (chr -1), pos 0, width = 1. <br /></td></tr>
<tr class="separator:aac6129647c2ad07c5ee63edb3f65c705"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adfb92846c32e6a877e60fc570827982e"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#adfb92846c32e6a877e60fc570827982e">GenomicRegion</a> (int32_t t_chr, int32_t t_pos1, int32_t t_pos2, char t_strand= '*')</td></tr>
<tr class="memdesc:adfb92846c32e6a877e60fc570827982e"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct a <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> at a specific start and end location.  <a href="#adfb92846c32e6a877e60fc570827982e">More...</a><br /></td></tr>
<tr class="separator:adfb92846c32e6a877e60fc570827982e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a58d938b810b33dd5f3a1af41ed501555"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a58d938b810b33dd5f3a1af41ed501555"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#a58d938b810b33dd5f3a1af41ed501555">GenomicRegion</a> (const std::string &amp;tchr, const std::string &amp;tpos1, const std::string &amp;tpos2, bam_hdr_t *h=NULL)</td></tr>
<tr class="memdesc:a58d938b810b33dd5f3a1af41ed501555"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct a <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> from a set of strings. <br /></td></tr>
<tr class="separator:a58d938b810b33dd5f3a1af41ed501555"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5b652a2056b386d4ad4c5c3ad6b29f91"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#a5b652a2056b386d4ad4c5c3ad6b29f91">GenomicRegion</a> (const std::string &amp;reg, bam_hdr_t *h)</td></tr>
<tr class="memdesc:a5b652a2056b386d4ad4c5c3ad6b29f91"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct a <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> from a samtools style region string.  <a href="#a5b652a2056b386d4ad4c5c3ad6b29f91">More...</a><br /></td></tr>
<tr class="separator:a5b652a2056b386d4ad4c5c3ad6b29f91"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3e62c2bcdfd19724bcd0c095065d413d"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#a3e62c2bcdfd19724bcd0c095065d413d">pointString</a> () const </td></tr>
<tr class="memdesc:a3e62c2bcdfd19724bcd0c095065d413d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return a string representation of just the first base-pair e.g.  <a href="#a3e62c2bcdfd19724bcd0c095065d413d">More...</a><br /></td></tr>
<tr class="separator:a3e62c2bcdfd19724bcd0c095065d413d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad720200233bac1be2d4ddae047f00871"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#ad720200233bac1be2d4ddae047f00871">random</a> ()</td></tr>
<tr class="memdesc:ad720200233bac1be2d4ddae047f00871"><td class="mdescLeft">&#160;</td><td class="mdescRight">Randomize the position of this <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> on the genome.  <a href="#ad720200233bac1be2d4ddae047f00871">More...</a><br /></td></tr>
<tr class="separator:ad720200233bac1be2d4ddae047f00871"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8d985300b138b6c5556ab17ed4df3b38"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a8d985300b138b6c5556ab17ed4df3b38"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#a8d985300b138b6c5556ab17ed4df3b38">valid</a> () const </td></tr>
<tr class="memdesc:a8d985300b138b6c5556ab17ed4df3b38"><td class="mdescLeft">&#160;</td><td class="mdescRight">Returns true if chr id &gt;= 0, false otherwise. <br /></td></tr>
<tr class="separator:a8d985300b138b6c5556ab17ed4df3b38"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a479432127ee77145cc19d6a2d1590821"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a479432127ee77145cc19d6a2d1590821"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#a479432127ee77145cc19d6a2d1590821">isEmpty</a> () const </td></tr>
<tr class="memdesc:a479432127ee77145cc19d6a2d1590821"><td class="mdescLeft">&#160;</td><td class="mdescRight">Check if the <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> is empty (aka chr -1 and pos1=pos2=0) <br /></td></tr>
<tr class="separator:a479432127ee77145cc19d6a2d1590821"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a61c937e653f783c51ceb49f7df568099"><td class="memItemLeft" align="right" valign="top">int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#a61c937e653f783c51ceb49f7df568099">distanceBetweenStarts</a> (const <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &amp;gr) const </td></tr>
<tr class="memdesc:a61c937e653f783c51ceb49f7df568099"><td class="mdescLeft">&#160;</td><td class="mdescRight">Find the absolute distance between start of two <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects.  <a href="#a61c937e653f783c51ceb49f7df568099">More...</a><br /></td></tr>
<tr class="separator:a61c937e653f783c51ceb49f7df568099"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a383af84213e4d6a2efd050be9de5dedc"><td class="memItemLeft" align="right" valign="top">int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#a383af84213e4d6a2efd050be9de5dedc">distanceBetweenEnds</a> (const <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &amp;gr) const </td></tr>
<tr class="memdesc:a383af84213e4d6a2efd050be9de5dedc"><td class="mdescLeft">&#160;</td><td class="mdescRight">Find the absolute distance between ends of two <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects.  <a href="#a383af84213e4d6a2efd050be9de5dedc">More...</a><br /></td></tr>
<tr class="separator:a383af84213e4d6a2efd050be9de5dedc"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a66f8ab72dec8e5ac02e5ce51e5525c15"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a66f8ab72dec8e5ac02e5ce51e5525c15"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><b>operator&lt;</b> (const <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &amp;b) const </td></tr>
<tr class="separator:a66f8ab72dec8e5ac02e5ce51e5525c15"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af2e7fd8b80b6054c0a9e7e8948a8ef20"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="af2e7fd8b80b6054c0a9e7e8948a8ef20"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><b>operator==</b> (const <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &amp;b) const </td></tr>
<tr class="separator:af2e7fd8b80b6054c0a9e7e8948a8ef20"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aff7c90c3e18cdb1e1f12f50eff380093"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aff7c90c3e18cdb1e1f12f50eff380093"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><b>operator&lt;=</b> (const <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &amp;b) const </td></tr>
<tr class="separator:aff7c90c3e18cdb1e1f12f50eff380093"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a914597c140483d04c34e0742501aae22"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#a914597c140483d04c34e0742501aae22">getOverlap</a> (const <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &amp;gr) const </td></tr>
<tr class="memdesc:a914597c140483d04c34e0742501aae22"><td class="mdescLeft">&#160;</td><td class="mdescRight">Check if the <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> has a complete or partial overlap If the argument contains the calling object, returns 3 If the argument is contained in the calling object, returns 2 If the argument overlaps partially the calling object, returns 1 If the argument and calling object do not overlap, returns 0.  <a href="#a914597c140483d04c34e0742501aae22">More...</a><br /></td></tr>
<tr class="separator:a914597c140483d04c34e0742501aae22"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3f23c97b6c9f2a88e1121d1c633ccafa"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a3f23c97b6c9f2a88e1121d1c633ccafa"></a>
std::string&#160;</td><td class="memItemRight" valign="bottom"><b>toString</b> () const </td></tr>
<tr class="separator:a3f23c97b6c9f2a88e1121d1c633ccafa"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aeac086a16e577f98a6b36707a2908680"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#aeac086a16e577f98a6b36707a2908680">ChrName</a> (const bam_hdr_t *h=nullptr) const </td></tr>
<tr class="memdesc:aeac086a16e577f98a6b36707a2908680"><td class="mdescLeft">&#160;</td><td class="mdescRight">Extract the chromosome name as a string.  <a href="#aeac086a16e577f98a6b36707a2908680">More...</a><br /></td></tr>
<tr class="separator:aeac086a16e577f98a6b36707a2908680"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6e48fac8d964c658610b53605ab511a1"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#a6e48fac8d964c658610b53605ab511a1">pad</a> (int32_t pad)</td></tr>
<tr class="memdesc:a6e48fac8d964c658610b53605ab511a1"><td class="mdescLeft">&#160;</td><td class="mdescRight">Pad the object to make larger or smaller.  <a href="#a6e48fac8d964c658610b53605ab511a1">More...</a><br /></td></tr>
<tr class="separator:a6e48fac8d964c658610b53605ab511a1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a369399896761e31ae71db57fdd0ba431"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a369399896761e31ae71db57fdd0ba431"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><b>width</b> () const </td></tr>
<tr class="separator:a369399896761e31ae71db57fdd0ba431"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-static-methods"></a>
Static Public Member Functions</h2></td></tr>
<tr class="memitem:a58916d0e2c8f87db60befe836d5995cf"><td class="memItemLeft" align="right" valign="top">static std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegion.html#a58916d0e2c8f87db60befe836d5995cf">chrToString</a> (int32_t ref)</td></tr>
<tr class="memdesc:a58916d0e2c8f87db60befe836d5995cf"><td class="mdescLeft">&#160;</td><td class="mdescRight">Convert a chromosome number to a string using default ordering (1-Y) Assumes a 1-based ordering (1, ...), not zero-based.  <a href="#a58916d0e2c8f87db60befe836d5995cf">More...</a><br /></td></tr>
<tr class="separator:a58916d0e2c8f87db60befe836d5995cf"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr class="memitem:aa3ddc44cfe3b601169e28bb7fbd84e90"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aa3ddc44cfe3b601169e28bb7fbd84e90"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><b>chr</b> = 0</td></tr>
<tr class="separator:aa3ddc44cfe3b601169e28bb7fbd84e90"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab09f4a0ed9bf23e212a6ed416d2d9de2"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ab09f4a0ed9bf23e212a6ed416d2d9de2"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><b>pos1</b> = 0</td></tr>
<tr class="separator:ab09f4a0ed9bf23e212a6ed416d2d9de2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a427e055801abb87fe04c6404983edec5"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a427e055801abb87fe04c6404983edec5"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><b>pos2</b> = 0</td></tr>
<tr class="separator:a427e055801abb87fe04c6404983edec5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae2a3cd2c3369fd69920d00953e304103"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ae2a3cd2c3369fd69920d00953e304103"></a>
char&#160;</td><td class="memItemRight" valign="bottom"><b>strand</b> = '*'</td></tr>
<tr class="separator:ae2a3cd2c3369fd69920d00953e304103"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="friends"></a>
Friends</h2></td></tr>
<tr class="memitem:a0f84df795d244124c4b5d529002c4c99"><td class="memTemplParams" colspan="2"><a class="anchor" id="a0f84df795d244124c4b5d529002c4c99"></a>
template&lt;typename T &gt; </td></tr>
<tr class="memitem:a0f84df795d244124c4b5d529002c4c99"><td class="memTemplItemLeft" align="right" valign="top">class&#160;</td><td class="memTemplItemRight" valign="bottom"><b>GenomicRegionCollection</b></td></tr>
<tr class="separator:a0f84df795d244124c4b5d529002c4c99"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a11718e48e706650b39944aa906194fd2"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a11718e48e706650b39944aa906194fd2"></a>
std::ostream &amp;&#160;</td><td class="memItemRight" valign="bottom"><b>operator&lt;&lt;</b> (std::ostream &amp;out, const <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &amp;gr)</td></tr>
<tr class="separator:a11718e48e706650b39944aa906194fd2"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Container for an interval on the genome. </p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="adfb92846c32e6a877e60fc570827982e"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> </td>
          <td>(</td>
          <td class="paramtype">int32_t&#160;</td>
          <td class="paramname"><em>t_chr</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int32_t&#160;</td>
          <td class="paramname"><em>t_pos1</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int32_t&#160;</td>
          <td class="paramname"><em>t_pos2</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">char&#160;</td>
          <td class="paramname"><em>t_strand</em> = <code>'*'</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Construct a <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> at a specific start and end location. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">t_chr</td><td>Chromosome id (chr1 = 0, etc) </td></tr>
    <tr><td class="paramname">t_pos1</td><td>Start position </td></tr>
    <tr><td class="paramname">t_pos2</td><td>End position. Must be &gt;= start position. </td></tr>
    <tr><td class="paramname">strand.</td><td>+, -, or * (default is *) </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">throws</td><td>an invalid_argument exception if pos2 &lt; pos1 </td></tr>
    <tr><td class="paramname">throws</td><td>an invalid_argument exception if char not one of +, - , * </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a5b652a2056b386d4ad4c5c3ad6b29f91"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>reg</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bam_hdr_t *&#160;</td>
          <td class="paramname"><em>h</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Construct a <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> from a samtools style region string. </p>
<p>This calls the samtools-like parser, which accepts in form "chr7:10,000-11,100". Note that this requires that a pointer to the BAM header be provided as well to convert the text representation of the chr to the id number. </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">reg</td><td>Samtools-style string (e.g. "1:1,000,000-2,000,000") </td></tr>
    <tr><td class="paramname">h</td><td>Pointer to BAM header that will be used to convert chr string to ref id </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">throws</td><td>an invalid_argument exception if cannot parse correctly </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="aeac086a16e577f98a6b36707a2908680"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string ChrName </td>
          <td>(</td>
          <td class="paramtype">const bam_hdr_t *&#160;</td>
          <td class="paramname"><em>h</em> = <code>nullptr</code></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Extract the chromosome name as a string. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">h</td><td>BAM header with h-&gt;target_name field </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">throws</td><td>an invalid_argument exception if ref id &gt;= h-&gt;n_targets </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a58916d0e2c8f87db60befe836d5995cf"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static std::string chrToString </td>
          <td>(</td>
          <td class="paramtype">int32_t&#160;</td>
          <td class="paramname"><em>ref</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Convert a chromosome number to a string using default ordering (1-Y) Assumes a 1-based ordering (1, ...), not zero-based. </p>
<p>e.g. chrToString(10) return "11" </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">ref</td><td>Reference ID to convert </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">throws</td><td>an invalid_argument exception if ref &lt; 0 </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a383af84213e4d6a2efd050be9de5dedc"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int32_t distanceBetweenEnds </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &amp;&#160;</td>
          <td class="paramname"><em>gr</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Find the absolute distance between ends of two <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects. </p>
<p>If chr1 != chr2, then -1 is returned </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">gr</td><td><a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> object to compare with </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a61c937e653f783c51ceb49f7df568099"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int32_t distanceBetweenStarts </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &amp;&#160;</td>
          <td class="paramname"><em>gr</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Find the absolute distance between start of two <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects. </p>
<p>If chr1 != chr2, then -1 is returned </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">gr</td><td><a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> object to compare with </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a914597c140483d04c34e0742501aae22"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int getOverlap </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &amp;&#160;</td>
          <td class="paramname"><em>gr</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Check if the <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> has a complete or partial overlap If the argument contains the calling object, returns 3 If the argument is contained in the calling object, returns 2 If the argument overlaps partially the calling object, returns 1 If the argument and calling object do not overlap, returns 0. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">gr</td><td><a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> to compare against </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a6e48fac8d964c658610b53605ab511a1"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void pad </td>
          <td>(</td>
          <td class="paramtype">int32_t&#160;</td>
          <td class="paramname"><em>pad</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Pad the object to make larger or smaller. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">pad</td><td>Amount to pad by. </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">throws</td><td>an out_of_bounds if for pad &lt; -width/2 </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a3e62c2bcdfd19724bcd0c095065d413d"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string pointString </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Return a string representation of just the first base-pair e.g. </p>
<p>1:10,000 </p>

</div>
</div>
<a class="anchor" id="ad720200233bac1be2d4ddae047f00871"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void random </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Randomize the position of this <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> on the genome. </p>
<p>Creates a <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> with pos1 = pos2. Simulates a random value with val &lt;= genome_size_XY and then converts to <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> </p><dl class="section note"><dt>Note</dt><dd>Seed is set before-hand at any time with srand </dd></dl>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>src/SnowTools/<a class="el" href="GenomicRegion_8h_source.html">GenomicRegion.h</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Fri Feb 12 2016 22:44:41 for SnowTools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.8
</small></address>
</body>
</html>
