<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.8"/>
<title>SnowTools: GenomicRegionCollection&lt; T &gt; Class Template Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">SnowTools
   &#160;<span id="projectnumber">0.0</span>
   </div>
   <div id="projectbrief">C++ htslib/bwa-mem interface and command line tools for interrogating sequence data</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.8 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="hierarchy.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceSnowTools.html">SnowTools</a></li><li class="navelem"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html">GenomicRegionCollection</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a> &#124;
<a href="classSnowTools_1_1GenomicRegionCollection-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">GenomicRegionCollection&lt; T &gt; Class Template Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>Template class for a collection of children of <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects.  
 <a href="classSnowTools_1_1GenomicRegionCollection.html#details">More...</a></p>

<p><code>#include &lt;<a class="el" href="GenomicRegionCollection_8h_source.html">GenomicRegionCollection.h</a>&gt;</code></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a09bc06d105cefe297c3077b0b52d95a8"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a09bc06d105cefe297c3077b0b52d95a8"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a09bc06d105cefe297c3077b0b52d95a8">GenomicRegionCollection</a> ()</td></tr>
<tr class="memdesc:a09bc06d105cefe297c3077b0b52d95a8"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct an empty <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a>. <br /></td></tr>
<tr class="separator:a09bc06d105cefe297c3077b0b52d95a8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa44ca278bece95440485c6a4b03523bd"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aa44ca278bece95440485c6a4b03523bd"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#aa44ca278bece95440485c6a4b03523bd">GenomicRegionCollection</a> (std::vector&lt; T &gt; &amp;vec)</td></tr>
<tr class="memdesc:aa44ca278bece95440485c6a4b03523bd"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct from a plain vector of <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects. <br /></td></tr>
<tr class="separator:aa44ca278bece95440485c6a4b03523bd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a762e8c86518753d99f092faaa00b8f70"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a762e8c86518753d99f092faaa00b8f70"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a762e8c86518753d99f092faaa00b8f70">GenomicRegionCollection</a> (const T &amp;gr)</td></tr>
<tr class="memdesc:a762e8c86518753d99f092faaa00b8f70"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct from a single <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. <br /></td></tr>
<tr class="separator:a762e8c86518753d99f092faaa00b8f70"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afd91845d9a19f319f262540d73e6cb01"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="afd91845d9a19f319f262540d73e6cb01"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#afd91845d9a19f319f262540d73e6cb01">GenomicRegionCollection</a> (const BamReadVector &amp;brv)</td></tr>
<tr class="memdesc:afd91845d9a19f319f262540d73e6cb01"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct from a vector of reads. <br /></td></tr>
<tr class="separator:afd91845d9a19f319f262540d73e6cb01"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a268ee90c740378ba89d69ad1b2ea792e"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a268ee90c740378ba89d69ad1b2ea792e">GenomicRegionCollection</a> (int <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a369399896761e31ae71db57fdd0ba431">width</a>, int ovlp, const T &amp;gr)</td></tr>
<tr class="memdesc:a268ee90c740378ba89d69ad1b2ea792e"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct a <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> with overlapping intervals.  <a href="#a268ee90c740378ba89d69ad1b2ea792e">More...</a><br /></td></tr>
<tr class="separator:a268ee90c740378ba89d69ad1b2ea792e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a97831bb9834e1114fb6d44493ef195c4"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a97831bb9834e1114fb6d44493ef195c4">readMuTect</a> (const std::string &amp;file, int <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a39b04fefb69cbda66f838ed08334f147">pad</a>=0, bam_hdr_t *h=NULL)</td></tr>
<tr class="memdesc:a97831bb9834e1114fb6d44493ef195c4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Read in a MuTect call-stats file and adds to <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> object.  <a href="#a97831bb9834e1114fb6d44493ef195c4">More...</a><br /></td></tr>
<tr class="separator:a97831bb9834e1114fb6d44493ef195c4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5fa9596e6265004aa63808e3da784c2f"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a5fa9596e6265004aa63808e3da784c2f">readBEDfile</a> (const std::string &amp;file, int <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a39b04fefb69cbda66f838ed08334f147">pad</a>=0, bam_hdr_t *h=NULL)</td></tr>
<tr class="memdesc:a5fa9596e6265004aa63808e3da784c2f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Read in a BED file and adds to <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> object.  <a href="#a5fa9596e6265004aa63808e3da784c2f">More...</a><br /></td></tr>
<tr class="separator:a5fa9596e6265004aa63808e3da784c2f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aafcd2aeb4ff41a9acc4064cf4349a27e"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#aafcd2aeb4ff41a9acc4064cf4349a27e">readVCFfile</a> (const std::string &amp;file, int <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a39b04fefb69cbda66f838ed08334f147">pad</a>=0, bam_hdr_t *h=NULL)</td></tr>
<tr class="memdesc:aafcd2aeb4ff41a9acc4064cf4349a27e"><td class="mdescLeft">&#160;</td><td class="mdescRight">Read in a VCF file and adds to <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> object.  <a href="#aafcd2aeb4ff41a9acc4064cf4349a27e">More...</a><br /></td></tr>
<tr class="separator:aafcd2aeb4ff41a9acc4064cf4349a27e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ace0e90a87a721ba840f6812aba4322f9"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#ace0e90a87a721ba840f6812aba4322f9">regionFileToGRV</a> (const std::string &amp;file, int <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a39b04fefb69cbda66f838ed08334f147">pad</a>=0, bam_hdr_t *h=NULL, bool chr_header=false)</td></tr>
<tr class="memdesc:ace0e90a87a721ba840f6812aba4322f9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Read in a text file (can be gzipped) and add to <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a>.  <a href="#ace0e90a87a721ba840f6812aba4322f9">More...</a><br /></td></tr>
<tr class="separator:ace0e90a87a721ba840f6812aba4322f9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a49dcb257a867a6a0fe6d7604419fd10c"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a49dcb257a867a6a0fe6d7604419fd10c">createTreeMap</a> ()</td></tr>
<tr class="memdesc:a49dcb257a867a6a0fe6d7604419fd10c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Fill in the GenomicIntervalTreeMap stored in this object.  <a href="#a49dcb257a867a6a0fe6d7604419fd10c">More...</a><br /></td></tr>
<tr class="separator:a49dcb257a867a6a0fe6d7604419fd10c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afcacb2e3364e61bd9bad7c6ba8ab3f24"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#afcacb2e3364e61bd9bad7c6ba8ab3f24">sendToBED</a> (const std::string file)</td></tr>
<tr class="memdesc:afcacb2e3364e61bd9bad7c6ba8ab3f24"><td class="mdescLeft">&#160;</td><td class="mdescRight">Send the <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> to a BED file.  <a href="#afcacb2e3364e61bd9bad7c6ba8ab3f24">More...</a><br /></td></tr>
<tr class="separator:afcacb2e3364e61bd9bad7c6ba8ab3f24"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a57e6549eb4e893fe2d601f6c90f86cfc"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a57e6549eb4e893fe2d601f6c90f86cfc"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a57e6549eb4e893fe2d601f6c90f86cfc">mergeOverlappingIntervals</a> ()</td></tr>
<tr class="memdesc:a57e6549eb4e893fe2d601f6c90f86cfc"><td class="mdescLeft">&#160;</td><td class="mdescRight">Reduces the <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects to minimal set. <br /></td></tr>
<tr class="separator:a57e6549eb4e893fe2d601f6c90f86cfc"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aac782da1f912bceb5d8ad00c8dc892ac"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aac782da1f912bceb5d8ad00c8dc892ac"></a>
size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#aac782da1f912bceb5d8ad00c8dc892ac">size</a> () const </td></tr>
<tr class="memdesc:aac782da1f912bceb5d8ad00c8dc892ac"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the number of GenomicRegions stored. <br /></td></tr>
<tr class="separator:aac782da1f912bceb5d8ad00c8dc892ac"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaf4097ed8a6175a3c348ddb9d3fbb14d"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aaf4097ed8a6175a3c348ddb9d3fbb14d"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#aaf4097ed8a6175a3c348ddb9d3fbb14d">add</a> (const T &amp;g)</td></tr>
<tr class="memdesc:aaf4097ed8a6175a3c348ddb9d3fbb14d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add a new <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> to end. <br /></td></tr>
<tr class="separator:aaf4097ed8a6175a3c348ddb9d3fbb14d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac6e61de369e994009e36f344f99c15ad"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ac6e61de369e994009e36f344f99c15ad"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#ac6e61de369e994009e36f344f99c15ad">empty</a> () const </td></tr>
<tr class="memdesc:ac6e61de369e994009e36f344f99c15ad"><td class="mdescLeft">&#160;</td><td class="mdescRight">Is this object empty? <br /></td></tr>
<tr class="separator:ac6e61de369e994009e36f344f99c15ad"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac8bb3912a3ce86b15842e79d0b421204"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ac8bb3912a3ce86b15842e79d0b421204"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#ac8bb3912a3ce86b15842e79d0b421204">clear</a> ()</td></tr>
<tr class="memdesc:ac8bb3912a3ce86b15842e79d0b421204"><td class="mdescLeft">&#160;</td><td class="mdescRight">Clear out all of the <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects. <br /></td></tr>
<tr class="separator:ac8bb3912a3ce86b15842e79d0b421204"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ade04a187707f1bb82b2dc8f9f9cfebea"><td class="memItemLeft" align="right" valign="top">const T &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#ade04a187707f1bb82b2dc8f9f9cfebea">at</a> (size_t i) const </td></tr>
<tr class="memdesc:ade04a187707f1bb82b2dc8f9f9cfebea"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieve a <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> at given index.  <a href="#ade04a187707f1bb82b2dc8f9f9cfebea">More...</a><br /></td></tr>
<tr class="separator:ade04a187707f1bb82b2dc8f9f9cfebea"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a46f1839d4ef53a9dfdc86cb7e2aba2b2"><td class="memItemLeft" align="right" valign="top">size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a46f1839d4ef53a9dfdc86cb7e2aba2b2">findOverlapping</a> (const T &amp;gr) const </td></tr>
<tr class="memdesc:a46f1839d4ef53a9dfdc86cb7e2aba2b2"><td class="mdescLeft">&#160;</td><td class="mdescRight">Find overlaps between this vector and input <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>.  <a href="#a46f1839d4ef53a9dfdc86cb7e2aba2b2">More...</a><br /></td></tr>
<tr class="separator:a46f1839d4ef53a9dfdc86cb7e2aba2b2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af0cdcdb2651c22a839c5f0915de553e3"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="af0cdcdb2651c22a839c5f0915de553e3"></a>
size_t&#160;</td><td class="memItemRight" valign="bottom"><b>countContained</b> (const T &amp;gr)</td></tr>
<tr class="separator:af0cdcdb2651c22a839c5f0915de553e3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adb7390afdb9c7edb5f70ca03f5e6855b"><td class="memTemplParams" colspan="2"><a class="anchor" id="adb7390afdb9c7edb5f70ca03f5e6855b"></a>
template&lt;class K &gt; </td></tr>
<tr class="memitem:adb7390afdb9c7edb5f70ca03f5e6855b"><td class="memTemplItemLeft" align="right" valign="top"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html">GenomicRegionCollection</a><br class="typebreak" />
&lt; <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &gt;&#160;</td><td class="memTemplItemRight" valign="bottom"><b>findOverlaps</b> (<a class="el" href="classSnowTools_1_1GenomicRegionCollection.html">GenomicRegionCollection</a>&lt; K &gt; &amp;subject, std::vector&lt; int32_t &gt; &amp;query_id, std::vector&lt; int32_t &gt; &amp;subject_id, bool ignore_strand=false) const </td></tr>
<tr class="separator:adb7390afdb9c7edb5f70ca03f5e6855b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a369399896761e31ae71db57fdd0ba431"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a369399896761e31ae71db57fdd0ba431"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a369399896761e31ae71db57fdd0ba431">width</a> () const </td></tr>
<tr class="memdesc:a369399896761e31ae71db57fdd0ba431"><td class="mdescLeft">&#160;</td><td class="mdescRight">The total amount spanned by this collection. <br /></td></tr>
<tr class="separator:a369399896761e31ae71db57fdd0ba431"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a39b04fefb69cbda66f838ed08334f147"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a39b04fefb69cbda66f838ed08334f147">pad</a> (int v)</td></tr>
<tr class="memdesc:a39b04fefb69cbda66f838ed08334f147"><td class="mdescLeft">&#160;</td><td class="mdescRight">Increase the left and right ends of each contained <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> by the pad value.  <a href="#a39b04fefb69cbda66f838ed08334f147">More...</a><br /></td></tr>
<tr class="separator:a39b04fefb69cbda66f838ed08334f147"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5d56cb4cffc7089978d354304eb5a7a6"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a5d56cb4cffc7089978d354304eb5a7a6"></a>
T &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a5d56cb4cffc7089978d354304eb5a7a6">operator[]</a> (size_t i)</td></tr>
<tr class="memdesc:a5d56cb4cffc7089978d354304eb5a7a6"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the i'th <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. <br /></td></tr>
<tr class="separator:a5d56cb4cffc7089978d354304eb5a7a6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a088d11cfed6e7783e6737ca67188cd61"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a088d11cfed6e7783e6737ca67188cd61"></a>
const T &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a088d11cfed6e7783e6737ca67188cd61">operator[]</a> (size_t i) const </td></tr>
<tr class="memdesc:a088d11cfed6e7783e6737ca67188cd61"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retreive the i'th <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. <br /></td></tr>
<tr class="separator:a088d11cfed6e7783e6737ca67188cd61"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaf762880dc4539a09394816c32e40596"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aaf762880dc4539a09394816c32e40596"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#aaf762880dc4539a09394816c32e40596">concat</a> (const <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html">GenomicRegionCollection</a>&lt; T &gt; &amp;g)</td></tr>
<tr class="memdesc:aaf762880dc4539a09394816c32e40596"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add two <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> objects together. <br /></td></tr>
<tr class="separator:aaf762880dc4539a09394816c32e40596"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a910badaf8d3abbfad021311302de6dde"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a910badaf8d3abbfad021311302de6dde">sendToBED</a> () const </td></tr>
<tr class="memdesc:a910badaf8d3abbfad021311302de6dde"><td class="mdescLeft">&#160;</td><td class="mdescRight">Output the <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> to a BED format.  <a href="#a910badaf8d3abbfad021311302de6dde">More...</a><br /></td></tr>
<tr class="separator:a910badaf8d3abbfad021311302de6dde"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a99d9ab26bc6b51d576f9d6ad4980e4d8"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a99d9ab26bc6b51d576f9d6ad4980e4d8">getNextGenomicRegion</a> (T &amp;gr)</td></tr>
<tr class="memdesc:a99d9ab26bc6b51d576f9d6ad4980e4d8"><td class="mdescLeft">&#160;</td><td class="mdescRight">Fill the next <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> object.  <a href="#a99d9ab26bc6b51d576f9d6ad4980e4d8">More...</a><br /></td></tr>
<tr class="separator:a99d9ab26bc6b51d576f9d6ad4980e4d8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3c5011cec28d87ec5f85ebde0c77c26a"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a3c5011cec28d87ec5f85ebde0c77c26a"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>gsort</b> ()</td></tr>
<tr class="separator:a3c5011cec28d87ec5f85ebde0c77c26a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a74966b52e880770cc74ecc8aa14a8f35"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a74966b52e880770cc74ecc8aa14a8f35"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a74966b52e880770cc74ecc8aa14a8f35">SortAndStretchRight</a> (int max)</td></tr>
<tr class="memdesc:a74966b52e880770cc74ecc8aa14a8f35"><td class="mdescLeft">&#160;</td><td class="mdescRight">Expand all the elements so they are sorted and become adjacent by stretching them to the right up to max. <br /></td></tr>
<tr class="separator:a74966b52e880770cc74ecc8aa14a8f35"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6ac1e0ee7757d8a02d8c8168539836c9"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a6ac1e0ee7757d8a02d8c8168539836c9"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#a6ac1e0ee7757d8a02d8c8168539836c9">SortAndStretchLeft</a> (int min)</td></tr>
<tr class="memdesc:a6ac1e0ee7757d8a02d8c8168539836c9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Expand all the elements so they are sorted and become adjacent by stretching them to the left down to min. <br /></td></tr>
<tr class="separator:a6ac1e0ee7757d8a02d8c8168539836c9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab8734e666421c9fe3b6380a818c6c727"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ab8734e666421c9fe3b6380a818c6c727"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html#ab8734e666421c9fe3b6380a818c6c727">rewind</a> ()</td></tr>
<tr class="memdesc:ab8734e666421c9fe3b6380a818c6c727"><td class="mdescLeft">&#160;</td><td class="mdescRight">Rewind the element pointer to the first <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. <br /></td></tr>
<tr class="separator:ab8734e666421c9fe3b6380a818c6c727"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4dda4a55a31cb2efe8927c80a4dfd75f"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a4dda4a55a31cb2efe8927c80a4dfd75f"></a>
GenomicRegionVector&#160;</td><td class="memItemRight" valign="bottom"><b>asGenomicRegionVector</b> () const </td></tr>
<tr class="separator:a4dda4a55a31cb2efe8927c80a4dfd75f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae49c361f230a136b5b1221631fc48727"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ae49c361f230a136b5b1221631fc48727"></a>
std::vector&lt; T &gt;::iterator&#160;</td><td class="memItemRight" valign="bottom"><b>begin</b> ()</td></tr>
<tr class="separator:ae49c361f230a136b5b1221631fc48727"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad2020c437b7b1e81d6f191d31ae8a2e2"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ad2020c437b7b1e81d6f191d31ae8a2e2"></a>
std::vector&lt; T &gt;::iterator&#160;</td><td class="memItemRight" valign="bottom"><b>end</b> ()</td></tr>
<tr class="separator:ad2020c437b7b1e81d6f191d31ae8a2e2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5be184186da9604378ff5080e26cbd73"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a5be184186da9604378ff5080e26cbd73"></a>
<a class="el" href="classSnowTools_1_1GenomicRegionCollection.html">GenomicRegionCollection</a><br class="typebreak" />
&lt; <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><b>intersection</b> (<a class="el" href="classSnowTools_1_1GenomicRegionCollection.html">GenomicRegionCollection</a>&lt; <a class="el" href="classSnowTools_1_1GenomicRegion.html">GenomicRegion</a> &gt; &amp;subject, bool ignore_strand=false)</td></tr>
<tr class="separator:a5be184186da9604378ff5080e26cbd73"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr class="memitem:ad9b0898c2a4ff440c840fceb8fde1208"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ad9b0898c2a4ff440c840fceb8fde1208"></a>
GenomicIntervalTreeMap&#160;</td><td class="memItemRight" valign="bottom"><b>m_tree</b></td></tr>
<tr class="separator:ad9b0898c2a4ff440c840fceb8fde1208"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a52c56006c05d9e4b8d1decbab36ea7b6"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a52c56006c05d9e4b8d1decbab36ea7b6"></a>
std::vector&lt; T &gt;&#160;</td><td class="memItemRight" valign="bottom"><b>m_grv</b></td></tr>
<tr class="separator:a52c56006c05d9e4b8d1decbab36ea7b6"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><h3>template&lt;typename T = GenomicRegion&gt;<br />
class SnowTools::GenomicRegionCollection&lt; T &gt;</h3>

<p>Template class for a collection of children of <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects. </p>
<p>Contains a set of <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects (or their children), and allows interval queries, reading/writing BED files, etc. </p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a268ee90c740378ba89d69ad1b2ea792e"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classSnowTools_1_1GenomicRegionCollection.html">GenomicRegionCollection</a> </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>width</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>ovlp</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const T &amp;&#160;</td>
          <td class="paramname"><em>gr</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Construct a <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> with overlapping intervals. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">width</td><td>Desired bin width </td></tr>
    <tr><td class="paramname">ovlp</td><td>Amount that the bins should overlap </td></tr>
    <tr><td class="paramname">gr</td><td><a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> to divide into smaller overlapping bins </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="ade04a187707f1bb82b2dc8f9f9cfebea"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const T&amp; at </td>
          <td>(</td>
          <td class="paramtype">size_t&#160;</td>
          <td class="paramname"><em>i</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieve a <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> at given index. </p>
<p>Note that this does not move the idx iterator, which is used to loop through all the regions. Throws an exception if the index is out of bounds. </p><dl class="section return"><dt>Returns</dt><dd><a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> pointed to by index i </dd></dl>

</div>
</div>
<a class="anchor" id="a49dcb257a867a6a0fe6d7604419fd10c"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void createTreeMap </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Fill in the GenomicIntervalTreeMap stored in this object. </p>
<p>A GenomicIntervalTreeMap is an unordered_map of GenomicIntervalTrees for each chromosome. A GenomicIntervalTree is an interval tree on the ranges defined by the genomic interval, with cargo set at the same <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> object. </p>

</div>
</div>
<a class="anchor" id="a46f1839d4ef53a9dfdc86cb7e2aba2b2"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">size_t findOverlapping </td>
          <td>(</td>
          <td class="paramtype">const T &amp;&#160;</td>
          <td class="paramname"><em>gr</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Find overlaps between this vector and input <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. </p>
<p>Requires that the GenomicIntervalTreeMap have been created first </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">gr</td><td>Region to test </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>Number of overlapping elements in this <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> </dd></dl>

</div>
</div>
<a class="anchor" id="a99d9ab26bc6b51d576f9d6ad4980e4d8"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">bool getNextGenomicRegion </td>
          <td>(</td>
          <td class="paramtype">T &amp;&#160;</td>
          <td class="paramname"><em>gr</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Fill the next <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> object. </p>
<dl class="section return"><dt>Returns</dt><dd>false if add end of vector </dd></dl>

</div>
</div>
<a class="anchor" id="a39b04fefb69cbda66f838ed08334f147"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void pad </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>v</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Increase the left and right ends of each contained <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> by the pad value. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">v</td><td>Amount to pad each end by. Result is increase in width by 2*pad. </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a5fa9596e6265004aa63808e3da784c2f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void readBEDfile </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>pad</em> = <code>0</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bam_hdr_t *&#160;</td>
          <td class="paramname"><em>h</em> = <code>NULL</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Read in a BED file and adds to <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> object. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">file</td><td>Path to BED file </td></tr>
    <tr><td class="paramname">pad</td><td>Amount to pad intervals by </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a97831bb9834e1114fb6d44493ef195c4"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void readMuTect </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>pad</em> = <code>0</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bam_hdr_t *&#160;</td>
          <td class="paramname"><em>h</em> = <code>NULL</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Read in a MuTect call-stats file and adds to <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> object. </p>
<p>Reads a MuTect call-stats file and imports only lines with KEEP marked. </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">file</td><td>Path to call-stats file </td></tr>
    <tr><td class="paramname">pad</td><td>Amount to pad intervals by </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="aafcd2aeb4ff41a9acc4064cf4349a27e"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void readVCFfile </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>pad</em> = <code>0</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bam_hdr_t *&#160;</td>
          <td class="paramname"><em>h</em> = <code>NULL</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Read in a VCF file and adds to <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> object. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">file</td><td>Path to BED file </td></tr>
    <tr><td class="paramname">pad</td><td>Amount to pad intervals by </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ace0e90a87a721ba840f6812aba4322f9"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void regionFileToGRV </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>pad</em> = <code>0</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bam_hdr_t *&#160;</td>
          <td class="paramname"><em>h</em> = <code>NULL</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>chr_header</em> = <code>false</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Read in a text file (can be gzipped) and add to <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a>. </p>
<p>This function will automatically detect which file type is being input: &ndash; ends in .vcf -&gt; readVCFfile &ndash; ends in .bed -&gt; readBEDfile &ndash; header contains "MuTect" -&gt; readMuTect The values are appended to existing vector of <a class="el" href="classSnowTools_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">file</td><td>Text file to read and store intervals </td></tr>
    <tr><td class="paramname">pad</td><td>Amount to pad the intervals by (calls <a class="el" href="classSnowTools_1_1GenomicRegion.html#a6e48fac8d964c658610b53605ab511a1" title="Pad the object to make larger or smaller. ">GenomicRegion::pad</a>) </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="afcacb2e3364e61bd9bad7c6ba8ab3f24"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void sendToBED </td>
          <td>(</td>
          <td class="paramtype">const std::string&#160;</td>
          <td class="paramname"><em>file</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Send the <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> to a BED file. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">file</td><td>Output file </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a910badaf8d3abbfad021311302de6dde"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string sendToBED </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Output the <a class="el" href="classSnowTools_1_1GenomicRegionCollection.html" title="Template class for a collection of children of GenomicRegion objects. ">GenomicRegionCollection</a> to a BED format. </p>
<p>Currently just outputs chr pos pos2 with no header. </p><dl class="section return"><dt>Returns</dt><dd>BED formated string reprsentation </dd></dl>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>src/SnowTools/<a class="el" href="GenomicRegionCollection_8h_source.html">GenomicRegionCollection.h</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sun Feb 14 2016 22:17:02 for SnowTools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.8
</small></address>
</body>
</html>
