<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.8"/>
<title>SeqLib: GenomicRegionCollection&lt; T &gt; Class Template Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">SeqLib
   &#160;<span id="projectnumber">0.0</span>
   </div>
   <div id="projectbrief">C++ htslib/bwa-mem interface and command line tools for interrogating sequence data</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.8 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="hierarchy.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><b>SeqLib</b></li><li class="navelem"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="classSeqLib_1_1GenomicRegionCollection-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">GenomicRegionCollection&lt; T &gt; Class Template Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>Template class to store / query a collection of genomic intervals.  
 <a href="classSeqLib_1_1GenomicRegionCollection.html#details">More...</a></p>

<p><code>#include &lt;<a class="el" href="GenomicRegionCollection_8h_source.html">GenomicRegionCollection.h</a>&gt;</code></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a09bc06d105cefe297c3077b0b52d95a8"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a09bc06d105cefe297c3077b0b52d95a8"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a09bc06d105cefe297c3077b0b52d95a8">GenomicRegionCollection</a> ()</td></tr>
<tr class="memdesc:a09bc06d105cefe297c3077b0b52d95a8"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct an empty <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a>. <br /></td></tr>
<tr class="separator:a09bc06d105cefe297c3077b0b52d95a8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa44ca278bece95440485c6a4b03523bd"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aa44ca278bece95440485c6a4b03523bd"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#aa44ca278bece95440485c6a4b03523bd">GenomicRegionCollection</a> (std::vector&lt; T &gt; &amp;vec)</td></tr>
<tr class="memdesc:aa44ca278bece95440485c6a4b03523bd"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct from a plain vector of <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects. <br /></td></tr>
<tr class="separator:aa44ca278bece95440485c6a4b03523bd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a762e8c86518753d99f092faaa00b8f70"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a762e8c86518753d99f092faaa00b8f70"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a762e8c86518753d99f092faaa00b8f70">GenomicRegionCollection</a> (const T &amp;gr)</td></tr>
<tr class="memdesc:a762e8c86518753d99f092faaa00b8f70"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct from a single <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. <br /></td></tr>
<tr class="separator:a762e8c86518753d99f092faaa00b8f70"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6651453f024285d14d3b7e23952cefe3"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a6651453f024285d14d3b7e23952cefe3">GenomicRegionCollection</a> (const BamRecordVector &amp;brv)</td></tr>
<tr class="memdesc:a6651453f024285d14d3b7e23952cefe3"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct from a vector of reads.  <a href="#a6651453f024285d14d3b7e23952cefe3">More...</a><br /></td></tr>
<tr class="separator:a6651453f024285d14d3b7e23952cefe3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a268ee90c740378ba89d69ad1b2ea792e"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a268ee90c740378ba89d69ad1b2ea792e">GenomicRegionCollection</a> (int width, int ovlp, const T &amp;gr)</td></tr>
<tr class="memdesc:a268ee90c740378ba89d69ad1b2ea792e"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct a <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a> with overlapping intervals.  <a href="#a268ee90c740378ba89d69ad1b2ea792e">More...</a><br /></td></tr>
<tr class="separator:a268ee90c740378ba89d69ad1b2ea792e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac44abf839cd3047c9f62882096646e43"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#ac44abf839cd3047c9f62882096646e43">GenomicRegionCollection</a> (int width, int ovlp, const HeaderSequenceVector &amp;h)</td></tr>
<tr class="memdesc:ac44abf839cd3047c9f62882096646e43"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct a tiled set of intervals across a genome.  <a href="#ac44abf839cd3047c9f62882096646e43">More...</a><br /></td></tr>
<tr class="separator:ac44abf839cd3047c9f62882096646e43"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad93af037ad754a81296c5a4a2111b030"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#ad93af037ad754a81296c5a4a2111b030">ReadBED</a> (const std::string &amp;file, const <a class="el" href="classSeqLib_1_1BamHeader.html">SeqLib::BamHeader</a> &amp;hdr)</td></tr>
<tr class="memdesc:ad93af037ad754a81296c5a4a2111b030"><td class="mdescLeft">&#160;</td><td class="mdescRight">Read in a BED file and adds to <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a> object.  <a href="#ad93af037ad754a81296c5a4a2111b030">More...</a><br /></td></tr>
<tr class="separator:ad93af037ad754a81296c5a4a2111b030"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a24ed3852f541c0449d885d24222b11fb"><td class="memItemLeft" align="right" valign="top">bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a24ed3852f541c0449d885d24222b11fb">ReadVCF</a> (const std::string &amp;file, const <a class="el" href="classSeqLib_1_1BamHeader.html">SeqLib::BamHeader</a> &amp;hdr)</td></tr>
<tr class="memdesc:a24ed3852f541c0449d885d24222b11fb"><td class="mdescLeft">&#160;</td><td class="mdescRight">Read in a VCF file and adds to <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a> object.  <a href="#a24ed3852f541c0449d885d24222b11fb">More...</a><br /></td></tr>
<tr class="separator:a24ed3852f541c0449d885d24222b11fb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2697cc8b37d7bca7c055228382a9b208"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a2697cc8b37d7bca7c055228382a9b208"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a2697cc8b37d7bca7c055228382a9b208">Shuffle</a> ()</td></tr>
<tr class="memdesc:a2697cc8b37d7bca7c055228382a9b208"><td class="mdescLeft">&#160;</td><td class="mdescRight">Shuffle the order of the intervals. <br /></td></tr>
<tr class="separator:a2697cc8b37d7bca7c055228382a9b208"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9af0c2d255d52cc1038b19baeaba5f12"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a9af0c2d255d52cc1038b19baeaba5f12">GenomicRegionCollection</a> (const std::string &amp;file, const <a class="el" href="classSeqLib_1_1BamHeader.html">BamHeader</a> &amp;hdr)</td></tr>
<tr class="memdesc:a9af0c2d255d52cc1038b19baeaba5f12"><td class="mdescLeft">&#160;</td><td class="mdescRight">Read in a text file (can be gzipped) and construct a <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a>.  <a href="#a9af0c2d255d52cc1038b19baeaba5f12">More...</a><br /></td></tr>
<tr class="separator:a9af0c2d255d52cc1038b19baeaba5f12"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac2fee81427145db943d6d7cff0492cba"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#ac2fee81427145db943d6d7cff0492cba">CreateTreeMap</a> ()</td></tr>
<tr class="memdesc:ac2fee81427145db943d6d7cff0492cba"><td class="mdescLeft">&#160;</td><td class="mdescRight">Create the set of interval trees (one per chromosome)  <a href="#ac2fee81427145db943d6d7cff0492cba">More...</a><br /></td></tr>
<tr class="separator:ac2fee81427145db943d6d7cff0492cba"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a32803f8a41866df9444927fd7f8ad87c"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a32803f8a41866df9444927fd7f8ad87c">MergeOverlappingIntervals</a> ()</td></tr>
<tr class="memdesc:a32803f8a41866df9444927fd7f8ad87c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Reduces the <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects to minimal set by merging overlapping intervals.  <a href="#a32803f8a41866df9444927fd7f8ad87c">More...</a><br /></td></tr>
<tr class="separator:a32803f8a41866df9444927fd7f8ad87c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aac782da1f912bceb5d8ad00c8dc892ac"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aac782da1f912bceb5d8ad00c8dc892ac"></a>
size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#aac782da1f912bceb5d8ad00c8dc892ac">size</a> () const </td></tr>
<tr class="memdesc:aac782da1f912bceb5d8ad00c8dc892ac"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the number of GenomicRegions stored. <br /></td></tr>
<tr class="separator:aac782da1f912bceb5d8ad00c8dc892ac"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaf4097ed8a6175a3c348ddb9d3fbb14d"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aaf4097ed8a6175a3c348ddb9d3fbb14d"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#aaf4097ed8a6175a3c348ddb9d3fbb14d">add</a> (const T &amp;g)</td></tr>
<tr class="memdesc:aaf4097ed8a6175a3c348ddb9d3fbb14d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add a new <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> to end. <br /></td></tr>
<tr class="separator:aaf4097ed8a6175a3c348ddb9d3fbb14d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaef62c8761ddd1b95824fc57b1b277b5"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aaef62c8761ddd1b95824fc57b1b277b5"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#aaef62c8761ddd1b95824fc57b1b277b5">IsEmpty</a> () const </td></tr>
<tr class="memdesc:aaef62c8761ddd1b95824fc57b1b277b5"><td class="mdescLeft">&#160;</td><td class="mdescRight">Is this object empty? <br /></td></tr>
<tr class="separator:aaef62c8761ddd1b95824fc57b1b277b5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac8bb3912a3ce86b15842e79d0b421204"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ac8bb3912a3ce86b15842e79d0b421204"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#ac8bb3912a3ce86b15842e79d0b421204">clear</a> ()</td></tr>
<tr class="memdesc:ac8bb3912a3ce86b15842e79d0b421204"><td class="mdescLeft">&#160;</td><td class="mdescRight">Clear out all of the <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects. <br /></td></tr>
<tr class="separator:ac8bb3912a3ce86b15842e79d0b421204"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0344a8815a0e8dbe371effee45f6c026"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a0344a8815a0e8dbe371effee45f6c026"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a0344a8815a0e8dbe371effee45f6c026">NumTree</a> () const </td></tr>
<tr class="memdesc:a0344a8815a0e8dbe371effee45f6c026"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the number of trees (eg number of chromosomes, each with own tree. <br /></td></tr>
<tr class="separator:a0344a8815a0e8dbe371effee45f6c026"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2f8ac8b7d3893cbd5bc319b3376a54af"><td class="memTemplParams" colspan="2">template&lt;class K &gt; </td></tr>
<tr class="memitem:a2f8ac8b7d3893cbd5bc319b3376a54af"><td class="memTemplItemLeft" align="right" valign="top">std::vector&lt; int &gt;&#160;</td><td class="memTemplItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a2f8ac8b7d3893cbd5bc319b3376a54af">FindOverlappedIntervals</a> (const K &amp;gr, bool ignore_strand) const </td></tr>
<tr class="memdesc:a2f8ac8b7d3893cbd5bc319b3376a54af"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the IDs of all intervals that overlap with a query range.  <a href="#a2f8ac8b7d3893cbd5bc319b3376a54af">More...</a><br /></td></tr>
<tr class="separator:a2f8ac8b7d3893cbd5bc319b3376a54af"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aee2ddbfea25184d5dec2ac875d712ef1"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aee2ddbfea25184d5dec2ac875d712ef1"></a>
const GenomicIntervalTreeMap *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#aee2ddbfea25184d5dec2ac875d712ef1">GetTree</a> () const </td></tr>
<tr class="memdesc:aee2ddbfea25184d5dec2ac875d712ef1"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get a const pointer to the genomic interval tree map. <br /></td></tr>
<tr class="separator:aee2ddbfea25184d5dec2ac875d712ef1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ade04a187707f1bb82b2dc8f9f9cfebea"><td class="memItemLeft" align="right" valign="top">const T &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#ade04a187707f1bb82b2dc8f9f9cfebea">at</a> (size_t i) const </td></tr>
<tr class="memdesc:ade04a187707f1bb82b2dc8f9f9cfebea"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieve a <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> at given index.  <a href="#ade04a187707f1bb82b2dc8f9f9cfebea">More...</a><br /></td></tr>
<tr class="separator:ade04a187707f1bb82b2dc8f9f9cfebea"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a022016b5b81346c9ad7c93f2e6e9b664"><td class="memItemLeft" align="right" valign="top">size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a022016b5b81346c9ad7c93f2e6e9b664">CountOverlaps</a> (const T &amp;gr) const </td></tr>
<tr class="memdesc:a022016b5b81346c9ad7c93f2e6e9b664"><td class="mdescLeft">&#160;</td><td class="mdescRight">Find overlaps between this vector and input <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>.  <a href="#a022016b5b81346c9ad7c93f2e6e9b664">More...</a><br /></td></tr>
<tr class="separator:a022016b5b81346c9ad7c93f2e6e9b664"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0016f23acdd05dc6cdc1c2df8593a70c"><td class="memTemplParams" colspan="2"><a class="anchor" id="a0016f23acdd05dc6cdc1c2df8593a70c"></a>
template&lt;class K &gt; </td></tr>
<tr class="memitem:a0016f23acdd05dc6cdc1c2df8593a70c"><td class="memTemplItemLeft" align="right" valign="top">bool&#160;</td><td class="memTemplItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a0016f23acdd05dc6cdc1c2df8593a70c">OverlapSameInterval</a> (const K &amp;gr1, const K &amp;gr2) const </td></tr>
<tr class="memdesc:a0016f23acdd05dc6cdc1c2df8593a70c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Test if two intervals overlap the same element in the collection. <br /></td></tr>
<tr class="separator:a0016f23acdd05dc6cdc1c2df8593a70c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7078273bca0402480d2233fc6281eaa7"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a7078273bca0402480d2233fc6281eaa7"></a>
size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a7078273bca0402480d2233fc6281eaa7">CountContained</a> (const T &amp;gr)</td></tr>
<tr class="memdesc:a7078273bca0402480d2233fc6281eaa7"><td class="mdescLeft">&#160;</td><td class="mdescRight">Count the number of intervals in the collection contained in this range. <br /></td></tr>
<tr class="separator:a7078273bca0402480d2233fc6281eaa7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4ec3d9f9cd226e92e23a305870f6a85a"><td class="memTemplParams" colspan="2">template&lt;class K &gt; </td></tr>
<tr class="memitem:a4ec3d9f9cd226e92e23a305870f6a85a"><td class="memTemplItemLeft" align="right" valign="top"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a><br class="typebreak" />
&lt; <a class="el" href="classSeqLib_1_1GenomicRegion.html">GenomicRegion</a> &gt;&#160;</td><td class="memTemplItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a4ec3d9f9cd226e92e23a305870f6a85a">FindOverlaps</a> (<a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a>&lt; K &gt; &amp;subject, std::vector&lt; int32_t &gt; &amp;query_id, std::vector&lt; int32_t &gt; &amp;subject_id, bool ignore_strand) const </td></tr>
<tr class="memdesc:a4ec3d9f9cd226e92e23a305870f6a85a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the overlaps between the collection and the query collection.  <a href="#a4ec3d9f9cd226e92e23a305870f6a85a">More...</a><br /></td></tr>
<tr class="separator:a4ec3d9f9cd226e92e23a305870f6a85a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3234faad21e3cb1872ed05efa452235d"><td class="memTemplParams" colspan="2">template&lt;class K &gt; </td></tr>
<tr class="memitem:a3234faad21e3cb1872ed05efa452235d"><td class="memTemplItemLeft" align="right" valign="top"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a><br class="typebreak" />
&lt; <a class="el" href="classSeqLib_1_1GenomicRegion.html">GenomicRegion</a> &gt;&#160;</td><td class="memTemplItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a3234faad21e3cb1872ed05efa452235d">FindOverlaps</a> (const K &amp;gr, bool ignore_strand) const </td></tr>
<tr class="memdesc:a3234faad21e3cb1872ed05efa452235d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the overlaps between the collection and the query interval.  <a href="#a3234faad21e3cb1872ed05efa452235d">More...</a><br /></td></tr>
<tr class="separator:a3234faad21e3cb1872ed05efa452235d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7a56156f002a092a98f4d265c5f52cdf"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a7a56156f002a092a98f4d265c5f52cdf"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a7a56156f002a092a98f4d265c5f52cdf">TotalWidth</a> () const </td></tr>
<tr class="memdesc:a7a56156f002a092a98f4d265c5f52cdf"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the total amount spanned by this collection. <br /></td></tr>
<tr class="separator:a7a56156f002a092a98f4d265c5f52cdf"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abf69b78dc237ad6f6b0292a7a9824979"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#abf69b78dc237ad6f6b0292a7a9824979">Pad</a> (int v)</td></tr>
<tr class="memdesc:abf69b78dc237ad6f6b0292a7a9824979"><td class="mdescLeft">&#160;</td><td class="mdescRight">Increase the left and right ends of each contained <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> by the pad value.  <a href="#abf69b78dc237ad6f6b0292a7a9824979">More...</a><br /></td></tr>
<tr class="separator:abf69b78dc237ad6f6b0292a7a9824979"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5d56cb4cffc7089978d354304eb5a7a6"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a5d56cb4cffc7089978d354304eb5a7a6"></a>
T &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a5d56cb4cffc7089978d354304eb5a7a6">operator[]</a> (size_t i)</td></tr>
<tr class="memdesc:a5d56cb4cffc7089978d354304eb5a7a6"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the i'th <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. <br /></td></tr>
<tr class="separator:a5d56cb4cffc7089978d354304eb5a7a6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a088d11cfed6e7783e6737ca67188cd61"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a088d11cfed6e7783e6737ca67188cd61"></a>
const T &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a088d11cfed6e7783e6737ca67188cd61">operator[]</a> (size_t i) const </td></tr>
<tr class="memdesc:a088d11cfed6e7783e6737ca67188cd61"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retreive the i'th <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. <br /></td></tr>
<tr class="separator:a088d11cfed6e7783e6737ca67188cd61"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acf78fb9915edeeadc03a35962582f0ff"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="acf78fb9915edeeadc03a35962582f0ff"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#acf78fb9915edeeadc03a35962582f0ff">Concat</a> (const <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a>&lt; T &gt; &amp;g)</td></tr>
<tr class="memdesc:acf78fb9915edeeadc03a35962582f0ff"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add two <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a> objects together. <br /></td></tr>
<tr class="separator:acf78fb9915edeeadc03a35962582f0ff"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a89d5c9e5c6f8210b201823a2d42059d6"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a89d5c9e5c6f8210b201823a2d42059d6">AsBEDString</a> (const <a class="el" href="classSeqLib_1_1BamHeader.html">BamHeader</a> &amp;h) const </td></tr>
<tr class="memdesc:a89d5c9e5c6f8210b201823a2d42059d6"><td class="mdescLeft">&#160;</td><td class="mdescRight">Output the <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a> to a BED format.  <a href="#a89d5c9e5c6f8210b201823a2d42059d6">More...</a><br /></td></tr>
<tr class="separator:a89d5c9e5c6f8210b201823a2d42059d6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9ecd24c26251a71355f0197f96d193b4"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a9ecd24c26251a71355f0197f96d193b4"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a9ecd24c26251a71355f0197f96d193b4">CoordinateSort</a> ()</td></tr>
<tr class="memdesc:a9ecd24c26251a71355f0197f96d193b4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Coordinate sort the interval collection. <br /></td></tr>
<tr class="separator:a9ecd24c26251a71355f0197f96d193b4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a74966b52e880770cc74ecc8aa14a8f35"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a74966b52e880770cc74ecc8aa14a8f35">SortAndStretchRight</a> (int max)</td></tr>
<tr class="memdesc:a74966b52e880770cc74ecc8aa14a8f35"><td class="mdescLeft">&#160;</td><td class="mdescRight">Expand all the elements so they are sorted and become adjacent by stretching them to the right up to max.  <a href="#a74966b52e880770cc74ecc8aa14a8f35">More...</a><br /></td></tr>
<tr class="separator:a74966b52e880770cc74ecc8aa14a8f35"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6ac1e0ee7757d8a02d8c8168539836c9"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a6ac1e0ee7757d8a02d8c8168539836c9">SortAndStretchLeft</a> (int min)</td></tr>
<tr class="memdesc:a6ac1e0ee7757d8a02d8c8168539836c9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Expand all the elements so they are sorted and become adjacent by stretching them to the left down to min.  <a href="#a6ac1e0ee7757d8a02d8c8168539836c9">More...</a><br /></td></tr>
<tr class="separator:a6ac1e0ee7757d8a02d8c8168539836c9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a335bf3cb31a54b9c5383de9579f70398"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a335bf3cb31a54b9c5383de9579f70398"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a335bf3cb31a54b9c5383de9579f70398">Rewind</a> ()</td></tr>
<tr class="memdesc:a335bf3cb31a54b9c5383de9579f70398"><td class="mdescLeft">&#160;</td><td class="mdescRight">Rewind the element pointer to the first <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. <br /></td></tr>
<tr class="separator:a335bf3cb31a54b9c5383de9579f70398"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3dcd65d56879f81ef118d4e0aa69dcf4"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a3dcd65d56879f81ef118d4e0aa69dcf4"></a>
GenomicRegionVector&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a3dcd65d56879f81ef118d4e0aa69dcf4">AsGenomicRegionVector</a> () const </td></tr>
<tr class="memdesc:a3dcd65d56879f81ef118d4e0aa69dcf4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return elements as an STL vector of <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects. <br /></td></tr>
<tr class="separator:a3dcd65d56879f81ef118d4e0aa69dcf4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae49c361f230a136b5b1221631fc48727"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ae49c361f230a136b5b1221631fc48727"></a>
std::vector&lt; T &gt;::iterator&#160;</td><td class="memItemRight" valign="bottom"><b>begin</b> ()</td></tr>
<tr class="separator:ae49c361f230a136b5b1221631fc48727"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad2020c437b7b1e81d6f191d31ae8a2e2"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ad2020c437b7b1e81d6f191d31ae8a2e2"></a>
std::vector&lt; T &gt;::iterator&#160;</td><td class="memItemRight" valign="bottom"><b>end</b> ()</td></tr>
<tr class="separator:ad2020c437b7b1e81d6f191d31ae8a2e2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8ccbb2fc356d3827e2f691db90fc143a"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a8ccbb2fc356d3827e2f691db90fc143a"></a>
std::vector&lt; T &gt;::const_iterator&#160;</td><td class="memItemRight" valign="bottom"><b>begin</b> () const </td></tr>
<tr class="separator:a8ccbb2fc356d3827e2f691db90fc143a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a69ca673576aad25f37820798001aa1a1"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a69ca673576aad25f37820798001aa1a1"></a>
std::vector&lt; T &gt;::const_iterator&#160;</td><td class="memItemRight" valign="bottom"><b>end</b> () const </td></tr>
<tr class="separator:a69ca673576aad25f37820798001aa1a1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5d8519b6efdde26e8901000180bc7e96"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a5d8519b6efdde26e8901000180bc7e96"></a>
<a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a><br class="typebreak" />
&lt; <a class="el" href="classSeqLib_1_1GenomicRegion.html">GenomicRegion</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#a5d8519b6efdde26e8901000180bc7e96">Intersection</a> (<a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a>&lt; <a class="el" href="classSeqLib_1_1GenomicRegion.html">GenomicRegion</a> &gt; &amp;subject, bool ignore_strand) const </td></tr>
<tr class="memdesc:a5d8519b6efdde26e8901000180bc7e96"><td class="mdescLeft">&#160;</td><td class="mdescRight">Shortcut to FindOverlaps that just returns the intersecting regions without keeping track of the query / subject ids. <br /></td></tr>
<tr class="separator:a5d8519b6efdde26e8901000180bc7e96"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><h3>template&lt;typename T = GenomicRegion&gt;<br />
class SeqLib::GenomicRegionCollection&lt; T &gt;</h3>

<p>Template class to store / query a collection of genomic intervals. </p>
<p>Can hold a collection of <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects, or any object whose class is a child of <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. Contains an implementation of an interval tree (as provided by Erik Garrison) for fast interval queries. </p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a6651453f024285d14d3b7e23952cefe3"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a> </td>
          <td>(</td>
          <td class="paramtype">const BamRecordVector &amp;&#160;</td>
          <td class="paramname"><em>brv</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Construct from a vector of reads. </p>
<dl class="section note"><dt>Note</dt><dd>See <a class="el" href="classSeqLib_1_1BamRecord.html#a2e3f6367ba0aaf20701f23af4101e754" title="Return read as a GenomicRegion. ">BamRecord::AsGenomicRegion</a> </dd></dl>

</div>
</div>
<a class="anchor" id="a268ee90c740378ba89d69ad1b2ea792e"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a> </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>width</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>ovlp</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const T &amp;&#160;</td>
          <td class="paramname"><em>gr</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Construct a <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a> with overlapping intervals. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">width</td><td>Desired bin width </td></tr>
    <tr><td class="paramname">ovlp</td><td>Amount that the bins should overlap </td></tr>
    <tr><td class="paramname">gr</td><td><a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> to divide into smaller overlapping bins </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ac44abf839cd3047c9f62882096646e43"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a> </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>width</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>ovlp</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const HeaderSequenceVector &amp;&#160;</td>
          <td class="paramname"><em>h</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Construct a tiled set of intervals across a genome. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">width</td><td>Width of each interval tile </td></tr>
    <tr><td class="paramname">ovlp</td><td>Amount of overlap between neighboring tiles </td></tr>
    <tr><td class="paramname">h</td><td>Set of chromosomes and their lengths to build the tile on </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a9af0c2d255d52cc1038b19baeaba5f12"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a> </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="classSeqLib_1_1BamHeader.html">BamHeader</a> &amp;&#160;</td>
          <td class="paramname"><em>hdr</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Read in a text file (can be gzipped) and construct a <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a>. </p>
<p>This function will automatically detect which file type is being input: &ndash; ends in .vcf -&gt; readVCFfile &ndash; ends in .bed -&gt; readBEDfile &ndash; header contains "MuTect" -&gt; readMuTect The values are appended to existing vector of <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">file</td><td>Text file to read and store intervals </td></tr>
    <tr><td class="paramname">hdr</td><td><a class="el" href="classSeqLib_1_1BamHeader.html" title="Store a header to a BAM file. ">BamHeader</a> to serve as dictionary for chromosomes </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="a89d5c9e5c6f8210b201823a2d42059d6"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string AsBEDString </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classSeqLib_1_1BamHeader.html">BamHeader</a> &amp;&#160;</td>
          <td class="paramname"><em>h</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Output the <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a> to a BED format. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">h</td><td>Header to convert id to chromosome name </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>BED formated string reprsentation </dd></dl>

</div>
</div>
<a class="anchor" id="ade04a187707f1bb82b2dc8f9f9cfebea"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const T&amp; at </td>
          <td>(</td>
          <td class="paramtype">size_t&#160;</td>
          <td class="paramname"><em>i</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieve a <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> at given index. </p>
<p>Note that this does not move the idx iterator, which is used to loop through all the regions. Throws an exception if the index is out of bounds. </p><dl class="section return"><dt>Returns</dt><dd><a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> pointed to by index i </dd></dl>

</div>
</div>
<a class="anchor" id="a022016b5b81346c9ad7c93f2e6e9b664"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">size_t CountOverlaps </td>
          <td>(</td>
          <td class="paramtype">const T &amp;&#160;</td>
          <td class="paramname"><em>gr</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Find overlaps between this vector and input <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. </p>
<p>Requires that the GenomicIntervalTreeMap have been created first </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">gr</td><td>Region to test </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>Number of overlapping elements in this <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a> </dd></dl>

</div>
</div>
<a class="anchor" id="ac2fee81427145db943d6d7cff0492cba"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void CreateTreeMap </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Create the set of interval trees (one per chromosome) </p>
<p>A GenomicIntervalTreeMap is an unordered_map of GenomicIntervalTrees for each chromosome. A GenomicIntervalTree is an interval tree on the ranges defined by the genomic interval, with cargo set at the same <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> object. </p>

</div>
</div>
<a class="anchor" id="a2f8ac8b7d3893cbd5bc319b3376a54af"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt;int&gt; FindOverlappedIntervals </td>
          <td>(</td>
          <td class="paramtype">const K &amp;&#160;</td>
          <td class="paramname"><em>gr</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>ignore_strand</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Get the IDs of all intervals that overlap with a query range. </p>
<p>The IDs are created during CreateTreeMap, and are the position of the the individual intervals from the tree, in genomic order. e.g the first interval on chromosome 1 gets 0, the next one gets 1, etc. The returned IDs can then be used as lookups with [], as long as the collection is not altered in between. </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">gr</td><td>Query range to check overlaps against </td></tr>
    <tr><td class="paramname">ignore_strand</td><td>Should strandedness be ignore when doing overlaps </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>A vector of IDs of intervals in this collection that overlap with gr </dd></dl>

</div>
</div>
<a class="anchor" id="a4ec3d9f9cd226e92e23a305870f6a85a"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a>&lt;<a class="el" href="classSeqLib_1_1GenomicRegion.html">GenomicRegion</a>&gt; FindOverlaps </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a>&lt; K &gt; &amp;&#160;</td>
          <td class="paramname"><em>subject</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">std::vector&lt; int32_t &gt; &amp;&#160;</td>
          <td class="paramname"><em>query_id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">std::vector&lt; int32_t &gt; &amp;&#160;</td>
          <td class="paramname"><em>subject_id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>ignore_strand</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Return the overlaps between the collection and the query collection. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">gr</td><td>Query collection of intervals </td></tr>
    <tr><td class="paramname">ignore_strand</td><td>If true, won't exclude overlap if on different strand </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>A collection of overlapping intervals from this collection, trimmed to be contained </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Throws</td><td>a logic_error if this tree is non-empty, but the interval tree has not been made with CreateTreeMap inside the query collection </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a3234faad21e3cb1872ed05efa452235d"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classSeqLib_1_1GenomicRegionCollection.html">GenomicRegionCollection</a>&lt;<a class="el" href="classSeqLib_1_1GenomicRegion.html">GenomicRegion</a>&gt; FindOverlaps </td>
          <td>(</td>
          <td class="paramtype">const K &amp;&#160;</td>
          <td class="paramname"><em>gr</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">bool&#160;</td>
          <td class="paramname"><em>ignore_strand</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Return the overlaps between the collection and the query interval. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">gr</td><td>Query region </td></tr>
    <tr><td class="paramname">ignore_strand</td><td>If true, won't exclude overlap if on different strand </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>A collection of overlapping intervals from this collection, trimmed to be contained inside gr </dd></dl>

</div>
</div>
<a class="anchor" id="a32803f8a41866df9444927fd7f8ad87c"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void MergeOverlappingIntervals </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Reduces the <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> objects to minimal set by merging overlapping intervals. </p>
<dl class="section note"><dt>Note</dt><dd>This will merge intervals that touch. eg [4,6] and [6,8] </dd>
<dd>
This will also call <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html#ac2fee81427145db943d6d7cff0492cba" title="Create the set of interval trees (one per chromosome) ">CreateTreeMap()</a> at end to re-create the interval tree </dd></dl>

</div>
</div>
<a class="anchor" id="abf69b78dc237ad6f6b0292a7a9824979"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void Pad </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>v</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Increase the left and right ends of each contained <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a> by the pad value. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">v</td><td>Amount to pad each end by. Result is increase in width by 2*pad. </td></tr>
  </table>
  </dd>
</dl>
<dl class="section note"><dt>Note</dt><dd>See <a class="el" href="classSeqLib_1_1GenomicRegion.html#af6e248cec09c02f177da29d5afd50311" title="Pad the object to make larger or smaller. ">GenomicRegion::Pad</a> </dd></dl>

</div>
</div>
<a class="anchor" id="ad93af037ad754a81296c5a4a2111b030"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">bool ReadBED </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="classSeqLib_1_1BamHeader.html">SeqLib::BamHeader</a> &amp;&#160;</td>
          <td class="paramname"><em>hdr</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Read in a BED file and adds to <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a> object. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">file</td><td>Path to BED file </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>True if file was succesfully read </dd></dl>

</div>
</div>
<a class="anchor" id="a24ed3852f541c0449d885d24222b11fb"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">bool ReadVCF </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="classSeqLib_1_1BamHeader.html">SeqLib::BamHeader</a> &amp;&#160;</td>
          <td class="paramname"><em>hdr</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Read in a VCF file and adds to <a class="el" href="classSeqLib_1_1GenomicRegionCollection.html" title="Template class to store / query a collection of genomic intervals. ">GenomicRegionCollection</a> object. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">file</td><td>Path to VCF file. All elements will be width = 1 (just read start point) </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a6ac1e0ee7757d8a02d8c8168539836c9"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void SortAndStretchLeft </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>min</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Expand all the elements so they are sorted and become adjacent by stretching them to the left down to min. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">min</td><td>Element furthest to the left will be stretched to min. If set to &lt; 0, will not stretch furthest right element. </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Throws</td><td>an out_of_range if furthest left is &lt; min </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a74966b52e880770cc74ecc8aa14a8f35"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void SortAndStretchRight </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>max</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Expand all the elements so they are sorted and become adjacent by stretching them to the right up to max. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">max</td><td>Element furthest to the right will be stretched to max. If set to 0, will not stretch furthest right element. </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Throws</td><td>an out_of_range if furthest right position is &gt; max </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>SeqLib/<a class="el" href="GenomicRegionCollection_8h_source.html">GenomicRegionCollection.h</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Fri Sep 23 2016 17:19:01 for SeqLib by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.8
</small></address>
</body>
</html>
