<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.8"/>
<title>SeqLib: BamRecord Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">SeqLib
   &#160;<span id="projectnumber">0.0</span>
   </div>
   <div id="projectbrief">C++ htslib/bwa-mem interface and command line tools for interrogating sequence data</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.8 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="hierarchy.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><b>SeqLib</b></li><li class="navelem"><a class="el" href="classSeqLib_1_1BamRecord.html">BamRecord</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#friends">Friends</a> &#124;
<a href="classSeqLib_1_1BamRecord-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">BamRecord Class Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>Class to store and interact with a SAM alignment record.  
 <a href="classSeqLib_1_1BamRecord.html#details">More...</a></p>

<p><code>#include &lt;<a class="el" href="BamRecord_8h_source.html">BamRecord.h</a>&gt;</code></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a176dac15321681fbcc9f3b21bde3da16"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a176dac15321681fbcc9f3b21bde3da16">BamRecord</a> (const std::string &amp;name, const std::string &amp;seq, const <a class="el" href="classSeqLib_1_1GenomicRegion.html">GenomicRegion</a> *gr, const <a class="el" href="classSeqLib_1_1Cigar.html">Cigar</a> &amp;cig)</td></tr>
<tr class="memdesc:a176dac15321681fbcc9f3b21bde3da16"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct a <a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> manually from a name, sequence, cigar and location.  <a href="#a176dac15321681fbcc9f3b21bde3da16">More...</a><br /></td></tr>
<tr class="separator:a176dac15321681fbcc9f3b21bde3da16"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a02fd73d861ef2e4aabb38c0c9ff82947"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a02fd73d861ef2e4aabb38c0c9ff82947"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a02fd73d861ef2e4aabb38c0c9ff82947">init</a> ()</td></tr>
<tr class="memdesc:a02fd73d861ef2e4aabb38c0c9ff82947"><td class="mdescLeft">&#160;</td><td class="mdescRight">Construct an empty <a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> by calling bam_init1() <br /></td></tr>
<tr class="separator:a02fd73d861ef2e4aabb38c0c9ff82947"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a479432127ee77145cc19d6a2d1590821"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a479432127ee77145cc19d6a2d1590821"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a479432127ee77145cc19d6a2d1590821">isEmpty</a> () const </td></tr>
<tr class="memdesc:a479432127ee77145cc19d6a2d1590821"><td class="mdescLeft">&#160;</td><td class="mdescRight">Check if a read is empty (not initialized)  true if read was not initialized with any values. <br /></td></tr>
<tr class="separator:a479432127ee77145cc19d6a2d1590821"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af1d3aa3591fce81d05b5515154181c70"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#af1d3aa3591fce81d05b5515154181c70">assign</a> (bam1_t *a)</td></tr>
<tr class="memdesc:af1d3aa3591fce81d05b5515154181c70"><td class="mdescLeft">&#160;</td><td class="mdescRight">Explicitly pass a bam1_t to the <a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a>.  <a href="#af1d3aa3591fce81d05b5515154181c70">More...</a><br /></td></tr>
<tr class="separator:af1d3aa3591fce81d05b5515154181c70"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9f8ed101ca6570d23e48ca7e11415dc1"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a9f8ed101ca6570d23e48ca7e11415dc1"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a9f8ed101ca6570d23e48ca7e11415dc1">BamRecord</a> ()</td></tr>
<tr class="memdesc:a9f8ed101ca6570d23e48ca7e11415dc1"><td class="mdescLeft">&#160;</td><td class="mdescRight">Make a <a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> with no memory allocated and a null header. <br /></td></tr>
<tr class="separator:a9f8ed101ca6570d23e48ca7e11415dc1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a20e39d0a44668c55b436cd7fd7d534d5"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a20e39d0a44668c55b436cd7fd7d534d5"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a20e39d0a44668c55b436cd7fd7d534d5">ReverseFlag</a> () const </td></tr>
<tr class="memdesc:a20e39d0a44668c55b436cd7fd7d534d5"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> is aligned on reverse strand. <br /></td></tr>
<tr class="separator:a20e39d0a44668c55b436cd7fd7d534d5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adcfaba5e50f7a799cb8d4d89fa5afae6"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="adcfaba5e50f7a799cb8d4d89fa5afae6"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#adcfaba5e50f7a799cb8d4d89fa5afae6">MateReverseFlag</a> () const </td></tr>
<tr class="memdesc:adcfaba5e50f7a799cb8d4d89fa5afae6"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> has mate aligned on reverse strand. <br /></td></tr>
<tr class="separator:adcfaba5e50f7a799cb8d4d89fa5afae6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa514c3ef20f1e7d98eb5d373fb8da356"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aa514c3ef20f1e7d98eb5d373fb8da356"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#aa514c3ef20f1e7d98eb5d373fb8da356">Interchromosomal</a> () const </td></tr>
<tr class="memdesc:aa514c3ef20f1e7d98eb5d373fb8da356"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> has is an interchromosomal alignment. <br /></td></tr>
<tr class="separator:aa514c3ef20f1e7d98eb5d373fb8da356"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7e60fc868c1af3555fb8bc1e81d3078d"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a7e60fc868c1af3555fb8bc1e81d3078d"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a7e60fc868c1af3555fb8bc1e81d3078d">DuplicateFlag</a> () const </td></tr>
<tr class="memdesc:a7e60fc868c1af3555fb8bc1e81d3078d"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> is a duplicate. <br /></td></tr>
<tr class="separator:a7e60fc868c1af3555fb8bc1e81d3078d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7ae334e53a310a160b86b9b6168eacf1"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a7ae334e53a310a160b86b9b6168eacf1"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a7ae334e53a310a160b86b9b6168eacf1">SecondaryFlag</a> () const </td></tr>
<tr class="memdesc:a7ae334e53a310a160b86b9b6168eacf1"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> is a secondary alignment. <br /></td></tr>
<tr class="separator:a7ae334e53a310a160b86b9b6168eacf1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3bc357eeaca6b02cc503358a3acc0a13"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a3bc357eeaca6b02cc503358a3acc0a13"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a3bc357eeaca6b02cc503358a3acc0a13">PairedFlag</a> () const </td></tr>
<tr class="memdesc:a3bc357eeaca6b02cc503358a3acc0a13"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> is paired. <br /></td></tr>
<tr class="separator:a3bc357eeaca6b02cc503358a3acc0a13"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7ce8b14d32f3167126f830b6c58904ed"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a7ce8b14d32f3167126f830b6c58904ed">PairOrientation</a> () const </td></tr>
<tr class="memdesc:a7ce8b14d32f3167126f830b6c58904ed"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the relative pair orientations.  <a href="#a7ce8b14d32f3167126f830b6c58904ed">More...</a><br /></td></tr>
<tr class="separator:a7ce8b14d32f3167126f830b6c58904ed"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abd8b975557b92d9828c7b884d840ccab"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="abd8b975557b92d9828c7b884d840ccab"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#abd8b975557b92d9828c7b884d840ccab">QCFailFlag</a> () const </td></tr>
<tr class="memdesc:abd8b975557b92d9828c7b884d840ccab"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> is failed QC. <br /></td></tr>
<tr class="separator:abd8b975557b92d9828c7b884d840ccab"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad49ec8ab930027b45d5705c043726af5"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ad49ec8ab930027b45d5705c043726af5"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ad49ec8ab930027b45d5705c043726af5">MappedFlag</a> () const </td></tr>
<tr class="memdesc:ad49ec8ab930027b45d5705c043726af5"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> is mapped. <br /></td></tr>
<tr class="separator:ad49ec8ab930027b45d5705c043726af5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af9c4451e82b8058d7bff9d99c6de01dd"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="af9c4451e82b8058d7bff9d99c6de01dd"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#af9c4451e82b8058d7bff9d99c6de01dd">MateMappedFlag</a> () const </td></tr>
<tr class="memdesc:af9c4451e82b8058d7bff9d99c6de01dd"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> mate is mapped. <br /></td></tr>
<tr class="separator:af9c4451e82b8058d7bff9d99c6de01dd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad4ab6b24a51712e0a8ce89e492d1d38e"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ad4ab6b24a51712e0a8ce89e492d1d38e"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ad4ab6b24a51712e0a8ce89e492d1d38e">PairMappedFlag</a> () const </td></tr>
<tr class="memdesc:ad4ab6b24a51712e0a8ce89e492d1d38e"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> is mapped and mate is mapped and in pair. <br /></td></tr>
<tr class="separator:ad4ab6b24a51712e0a8ce89e492d1d38e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6c541b6a75842bd38174b64bc229b50f"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a6c541b6a75842bd38174b64bc229b50f"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a6c541b6a75842bd38174b64bc229b50f">ProperPair</a> () const </td></tr>
<tr class="memdesc:a6c541b6a75842bd38174b64bc229b50f"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> is mapped in proper pair. <br /></td></tr>
<tr class="separator:a6c541b6a75842bd38174b64bc229b50f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a838ab2f97e8b9eea1be40da16bd2b136"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a838ab2f97e8b9eea1be40da16bd2b136"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a838ab2f97e8b9eea1be40da16bd2b136">ProperOrientation</a> () const </td></tr>
<tr class="memdesc:a838ab2f97e8b9eea1be40da16bd2b136"><td class="mdescLeft">&#160;</td><td class="mdescRight"><a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> has proper orientation (FR) <br /></td></tr>
<tr class="separator:a838ab2f97e8b9eea1be40da16bd2b136"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0dcbc1e45f86d3c2b22701a7d6afa43b"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a0dcbc1e45f86d3c2b22701a7d6afa43b"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a0dcbc1e45f86d3c2b22701a7d6afa43b">CountNBases</a> () const </td></tr>
<tr class="memdesc:a0dcbc1e45f86d3c2b22701a7d6afa43b"><td class="mdescLeft">&#160;</td><td class="mdescRight">Count the total number of N bases in this sequence. <br /></td></tr>
<tr class="separator:a0dcbc1e45f86d3c2b22701a7d6afa43b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab479d342def7b30ecb7d15bc29c4cc12"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ab479d342def7b30ecb7d15bc29c4cc12">QualityTrimmedSequence</a> (int32_t qualTrim, int32_t &amp;startpoint, int32_t &amp;endpoint) const </td></tr>
<tr class="memdesc:ab479d342def7b30ecb7d15bc29c4cc12"><td class="mdescLeft">&#160;</td><td class="mdescRight">Trim the sequence down by removing bases from ends with low quality scores.  <a href="#ab479d342def7b30ecb7d15bc29c4cc12">More...</a><br /></td></tr>
<tr class="separator:ab479d342def7b30ecb7d15bc29c4cc12"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae09627143c815100d284c660962c5571"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ae09627143c815100d284c660962c5571">QualitySequence</a> () const </td></tr>
<tr class="memdesc:ae09627143c815100d284c660962c5571"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieve the quality trimmed seqeuence from QT tag if made.  <a href="#ae09627143c815100d284c660962c5571">More...</a><br /></td></tr>
<tr class="separator:ae09627143c815100d284c660962c5571"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6fd3f2218c3245356bd3f841616d208e"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a6fd3f2218c3245356bd3f841616d208e"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a6fd3f2218c3245356bd3f841616d208e">Position</a> () const </td></tr>
<tr class="memdesc:a6fd3f2218c3245356bd3f841616d208e"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the alignment position. <br /></td></tr>
<tr class="separator:a6fd3f2218c3245356bd3f841616d208e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6a602cda286b646dd00170fec4bdd353"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a6a602cda286b646dd00170fec4bdd353"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a6a602cda286b646dd00170fec4bdd353">MatePosition</a> () const </td></tr>
<tr class="memdesc:a6a602cda286b646dd00170fec4bdd353"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the alignment position of mate. <br /></td></tr>
<tr class="separator:a6a602cda286b646dd00170fec4bdd353"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae056a71f4a71fe2d81218b1d93d1cfbe"><td class="memItemLeft" align="right" valign="top">int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ae056a71f4a71fe2d81218b1d93d1cfbe">CountBWASecondaryAlignments</a> () const </td></tr>
<tr class="memdesc:ae056a71f4a71fe2d81218b1d93d1cfbe"><td class="mdescLeft">&#160;</td><td class="mdescRight">Count the number of secondary alignments by looking at XA tag.  <a href="#ae056a71f4a71fe2d81218b1d93d1cfbe">More...</a><br /></td></tr>
<tr class="separator:ae056a71f4a71fe2d81218b1d93d1cfbe"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1d062cfbf0c0a2f1faa7238f0d5c013b"><td class="memItemLeft" align="right" valign="top">int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a1d062cfbf0c0a2f1faa7238f0d5c013b">CountBWAChimericAlignments</a> () const </td></tr>
<tr class="memdesc:a1d062cfbf0c0a2f1faa7238f0d5c013b"><td class="mdescLeft">&#160;</td><td class="mdescRight">Count the number of chimeric alignments by looking at XP and SA tags.  <a href="#a1d062cfbf0c0a2f1faa7238f0d5c013b">More...</a><br /></td></tr>
<tr class="separator:a1d062cfbf0c0a2f1faa7238f0d5c013b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1674436fb351427ebcba8da38823b53d"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a1674436fb351427ebcba8da38823b53d"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a1674436fb351427ebcba8da38823b53d">PositionEnd</a> () const </td></tr>
<tr class="memdesc:a1674436fb351427ebcba8da38823b53d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the end of the alignment. <br /></td></tr>
<tr class="separator:a1674436fb351427ebcba8da38823b53d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afbd518132e9a953ba6b4059f273b5168"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="afbd518132e9a953ba6b4059f273b5168"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#afbd518132e9a953ba6b4059f273b5168">ChrID</a> () const </td></tr>
<tr class="memdesc:afbd518132e9a953ba6b4059f273b5168"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the chromosome ID of the read. <br /></td></tr>
<tr class="separator:afbd518132e9a953ba6b4059f273b5168"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a26c0c601f0843ebb0e3ad6523ec9ba79"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a26c0c601f0843ebb0e3ad6523ec9ba79"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a26c0c601f0843ebb0e3ad6523ec9ba79">MateChrID</a> () const </td></tr>
<tr class="memdesc:a26c0c601f0843ebb0e3ad6523ec9ba79"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the chrosome ID of the mate read. <br /></td></tr>
<tr class="separator:a26c0c601f0843ebb0e3ad6523ec9ba79"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a18669c497e243e6ee41e19bed2500b32"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a18669c497e243e6ee41e19bed2500b32"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a18669c497e243e6ee41e19bed2500b32">MapQuality</a> () const </td></tr>
<tr class="memdesc:a18669c497e243e6ee41e19bed2500b32"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the mapping quality. <br /></td></tr>
<tr class="separator:a18669c497e243e6ee41e19bed2500b32"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9600062f79fb330c7ee41ec7bfad3d8c"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a9600062f79fb330c7ee41ec7bfad3d8c"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a9600062f79fb330c7ee41ec7bfad3d8c">SetMapQuality</a> (int32_t m)</td></tr>
<tr class="memdesc:a9600062f79fb330c7ee41ec7bfad3d8c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the mapping quality. <br /></td></tr>
<tr class="separator:a9600062f79fb330c7ee41ec7bfad3d8c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae49fd3017f098576620b4d11eafeaf18"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ae49fd3017f098576620b4d11eafeaf18"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ae49fd3017f098576620b4d11eafeaf18">SetChrID</a> (int32_t i)</td></tr>
<tr class="memdesc:ae49fd3017f098576620b4d11eafeaf18"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the chr id. <br /></td></tr>
<tr class="separator:ae49fd3017f098576620b4d11eafeaf18"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a09fdd3e70ec3a7b67950ca19ac1c1acf"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a09fdd3e70ec3a7b67950ca19ac1c1acf"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a09fdd3e70ec3a7b67950ca19ac1c1acf">SetChrIDMate</a> (int32_t i)</td></tr>
<tr class="memdesc:a09fdd3e70ec3a7b67950ca19ac1c1acf"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the chr id of mate. <br /></td></tr>
<tr class="separator:a09fdd3e70ec3a7b67950ca19ac1c1acf"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a641c2c78e70e38a3654d20bb87510083"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a641c2c78e70e38a3654d20bb87510083"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a641c2c78e70e38a3654d20bb87510083">SetPositionMate</a> (int32_t i)</td></tr>
<tr class="memdesc:a641c2c78e70e38a3654d20bb87510083"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the position of the mate read. <br /></td></tr>
<tr class="separator:a641c2c78e70e38a3654d20bb87510083"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adeaf7792eb11d59c8830c3d47e032ef9"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="adeaf7792eb11d59c8830c3d47e032ef9"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#adeaf7792eb11d59c8830c3d47e032ef9">SetPairMappedFlag</a> ()</td></tr>
<tr class="memdesc:adeaf7792eb11d59c8830c3d47e032ef9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the pair mapped flag on. <br /></td></tr>
<tr class="separator:adeaf7792eb11d59c8830c3d47e032ef9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a882a2a53bddb8284acb2c3f38d9c6c8f"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a882a2a53bddb8284acb2c3f38d9c6c8f"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a882a2a53bddb8284acb2c3f38d9c6c8f">SetMateReverseFlag</a> ()</td></tr>
<tr class="memdesc:a882a2a53bddb8284acb2c3f38d9c6c8f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the mate reverse flag on. <br /></td></tr>
<tr class="separator:a882a2a53bddb8284acb2c3f38d9c6c8f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a27093dc5aa273054170b9f550038bb41"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a27093dc5aa273054170b9f550038bb41"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a27093dc5aa273054170b9f550038bb41">CigarSize</a> () const </td></tr>
<tr class="memdesc:a27093dc5aa273054170b9f550038bb41"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the number of cigar fields. <br /></td></tr>
<tr class="separator:a27093dc5aa273054170b9f550038bb41"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aca866c832b39259ef75ed58fefcbded9"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aca866c832b39259ef75ed58fefcbded9"></a>
bool&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#aca866c832b39259ef75ed58fefcbded9">FirstFlag</a> () const </td></tr>
<tr class="memdesc:aca866c832b39259ef75ed58fefcbded9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Check if this read is first in pair. <br /></td></tr>
<tr class="separator:aca866c832b39259ef75ed58fefcbded9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a98694bf297ed2b002773ededb9b7f0c9"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a98694bf297ed2b002773ededb9b7f0c9"></a>
std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a98694bf297ed2b002773ededb9b7f0c9">Qname</a> () const </td></tr>
<tr class="memdesc:a98694bf297ed2b002773ededb9b7f0c9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the qname of this read as a string. <br /></td></tr>
<tr class="separator:a98694bf297ed2b002773ededb9b7f0c9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7c168c49357c6ceb03cfb72b7b2d15e7"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a7c168c49357c6ceb03cfb72b7b2d15e7"></a>
char *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a7c168c49357c6ceb03cfb72b7b2d15e7">QnameChar</a> () const </td></tr>
<tr class="memdesc:a7c168c49357c6ceb03cfb72b7b2d15e7"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the qname of this read as a char array. <br /></td></tr>
<tr class="separator:a7c168c49357c6ceb03cfb72b7b2d15e7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2299e80acbc4b3613721a207983806ac"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a2299e80acbc4b3613721a207983806ac"></a>
uint32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a2299e80acbc4b3613721a207983806ac">AlignmentFlag</a> () const </td></tr>
<tr class="memdesc:a2299e80acbc4b3613721a207983806ac"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the full alignment flag for this read. <br /></td></tr>
<tr class="separator:a2299e80acbc4b3613721a207983806ac"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1850a3aa2bdd222c675a9af449151ed8"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a1850a3aa2bdd222c675a9af449151ed8"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a1850a3aa2bdd222c675a9af449151ed8">InsertSize</a> () const </td></tr>
<tr class="memdesc:a1850a3aa2bdd222c675a9af449151ed8"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the insert size for this read. <br /></td></tr>
<tr class="separator:a1850a3aa2bdd222c675a9af449151ed8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acf667c1781f31e6518ed7e5dedb8c61a"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#acf667c1781f31e6518ed7e5dedb8c61a">ParseReadGroup</a> () const </td></tr>
<tr class="memdesc:acf667c1781f31e6518ed7e5dedb8c61a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the read group, first from qname, then by RG tag.  <a href="#acf667c1781f31e6518ed7e5dedb8c61a">More...</a><br /></td></tr>
<tr class="separator:acf667c1781f31e6518ed7e5dedb8c61a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa4fffc6cb10c7d77ddbdae9ad295a830"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aa4fffc6cb10c7d77ddbdae9ad295a830"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#aa4fffc6cb10c7d77ddbdae9ad295a830">FullInsertSize</a> () const </td></tr>
<tr class="memdesc:aa4fffc6cb10c7d77ddbdae9ad295a830"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the insert size, absolute value, and always taking into account read length. <br /></td></tr>
<tr class="separator:aa4fffc6cb10c7d77ddbdae9ad295a830"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a407e59e0a5addbdeff41217027bbddb0"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a407e59e0a5addbdeff41217027bbddb0"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a407e59e0a5addbdeff41217027bbddb0">Length</a> () const </td></tr>
<tr class="memdesc:a407e59e0a5addbdeff41217027bbddb0"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the number of query bases of this read (aka length) <br /></td></tr>
<tr class="separator:a407e59e0a5addbdeff41217027bbddb0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a60e240793312c62580035d5eebc58ef2"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a60e240793312c62580035d5eebc58ef2"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a60e240793312c62580035d5eebc58ef2">SmartAddTag</a> (const std::string &amp;tag, const std::string &amp;val)</td></tr>
<tr class="memdesc:a60e240793312c62580035d5eebc58ef2"><td class="mdescLeft">&#160;</td><td class="mdescRight">Append a tag with new value, delimited by 'x'. <br /></td></tr>
<tr class="separator:a60e240793312c62580035d5eebc58ef2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a442309741df8c9c57a0ff91473b2e7ed"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a442309741df8c9c57a0ff91473b2e7ed"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a442309741df8c9c57a0ff91473b2e7ed">SetQname</a> (const std::string &amp;n)</td></tr>
<tr class="memdesc:a442309741df8c9c57a0ff91473b2e7ed"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the query name. <br /></td></tr>
<tr class="separator:a442309741df8c9c57a0ff91473b2e7ed"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6ac91f0c10ab833ee3198d45c7e23241"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a6ac91f0c10ab833ee3198d45c7e23241"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>SetQualities</b> (const std::string &amp;n, int offset)</td></tr>
<tr class="separator:a6ac91f0c10ab833ee3198d45c7e23241"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aca7f1f1a9f775d2a51daf9004fc48d29"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aca7f1f1a9f775d2a51daf9004fc48d29"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#aca7f1f1a9f775d2a51daf9004fc48d29">SetSequence</a> (const std::string &amp;seq)</td></tr>
<tr class="memdesc:aca7f1f1a9f775d2a51daf9004fc48d29"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the sequence name. <br /></td></tr>
<tr class="separator:aca7f1f1a9f775d2a51daf9004fc48d29"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a347008abb35676d666573a96095a3f6f"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a347008abb35676d666573a96095a3f6f"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a347008abb35676d666573a96095a3f6f">SetCigar</a> (const <a class="el" href="classSeqLib_1_1Cigar.html">Cigar</a> &amp;c)</td></tr>
<tr class="memdesc:a347008abb35676d666573a96095a3f6f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the cigar field explicitly. <br /></td></tr>
<tr class="separator:a347008abb35676d666573a96095a3f6f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2e3f6367ba0aaf20701f23af4101e754"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a2e3f6367ba0aaf20701f23af4101e754"></a>
<a class="el" href="classSeqLib_1_1GenomicRegion.html">GenomicRegion</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a2e3f6367ba0aaf20701f23af4101e754">AsGenomicRegion</a> () const </td></tr>
<tr class="memdesc:a2e3f6367ba0aaf20701f23af4101e754"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return read as a <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. <br /></td></tr>
<tr class="separator:a2e3f6367ba0aaf20701f23af4101e754"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aadd0b0c8b6c56cb3ea48ef13d6fc6c89"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aadd0b0c8b6c56cb3ea48ef13d6fc6c89"></a>
<a class="el" href="classSeqLib_1_1GenomicRegion.html">GenomicRegion</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#aadd0b0c8b6c56cb3ea48ef13d6fc6c89">AsGenomicRegionMate</a> () const </td></tr>
<tr class="memdesc:aadd0b0c8b6c56cb3ea48ef13d6fc6c89"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return mate read as a <a class="el" href="classSeqLib_1_1GenomicRegion.html" title="Container for an interval on the genome. ">GenomicRegion</a>. <br /></td></tr>
<tr class="separator:aadd0b0c8b6c56cb3ea48ef13d6fc6c89"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a663d6d827b5f26f89ee995afb814963a"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a663d6d827b5f26f89ee995afb814963a">NumAlignedBases</a> () const </td></tr>
<tr class="memdesc:a663d6d827b5f26f89ee995afb814963a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the number of "aligned bases" in the same style as BamTools.  <a href="#a663d6d827b5f26f89ee995afb814963a">More...</a><br /></td></tr>
<tr class="separator:a663d6d827b5f26f89ee995afb814963a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac015633382d1cc7901559502e196025a"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ac015633382d1cc7901559502e196025a"></a>
uint32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ac015633382d1cc7901559502e196025a">MaxInsertionBases</a> () const </td></tr>
<tr class="memdesc:ac015633382d1cc7901559502e196025a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the max single insertion size on this cigar. <br /></td></tr>
<tr class="separator:ac015633382d1cc7901559502e196025a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5b1bfd8a761030d2b08d8c73528b6544"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a5b1bfd8a761030d2b08d8c73528b6544"></a>
uint32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a5b1bfd8a761030d2b08d8c73528b6544">MaxDeletionBases</a> () const </td></tr>
<tr class="memdesc:a5b1bfd8a761030d2b08d8c73528b6544"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the max single deletion size on this cigar. <br /></td></tr>
<tr class="separator:a5b1bfd8a761030d2b08d8c73528b6544"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6deb081aeeb1003f623dc6ad09ff2c19"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a6deb081aeeb1003f623dc6ad09ff2c19"></a>
uint32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a6deb081aeeb1003f623dc6ad09ff2c19">NumMatchBases</a> () const </td></tr>
<tr class="memdesc:a6deb081aeeb1003f623dc6ad09ff2c19"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the number of matched bases in this alignment. <br /></td></tr>
<tr class="separator:a6deb081aeeb1003f623dc6ad09ff2c19"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac0b4c8655d307fa4fc630e7e8117c3a6"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ac0b4c8655d307fa4fc630e7e8117c3a6"></a>
<a class="el" href="classSeqLib_1_1Cigar.html">Cigar</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ac0b4c8655d307fa4fc630e7e8117c3a6">GetCigar</a> () const </td></tr>
<tr class="memdesc:ac0b4c8655d307fa4fc630e7e8117c3a6"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieve the CIGAR as a more managable <a class="el" href="classSeqLib_1_1Cigar.html" title="CIGAR for a single gapped alignment. ">Cigar</a> structure. <br /></td></tr>
<tr class="separator:ac0b4c8655d307fa4fc630e7e8117c3a6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a59c86bc15d8eee99c031b6dc8ca248b1"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a59c86bc15d8eee99c031b6dc8ca248b1"></a>
<a class="el" href="classSeqLib_1_1Cigar.html">Cigar</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a59c86bc15d8eee99c031b6dc8ca248b1">GetReverseCigar</a> () const </td></tr>
<tr class="memdesc:a59c86bc15d8eee99c031b6dc8ca248b1"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieve the inverse of the CIGAR as a more managable <a class="el" href="classSeqLib_1_1Cigar.html" title="CIGAR for a single gapped alignment. ">Cigar</a> structure. <br /></td></tr>
<tr class="separator:a59c86bc15d8eee99c031b6dc8ca248b1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9534f17fd0cfabf55455d733bb3715c2"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a9534f17fd0cfabf55455d733bb3715c2">ClearSeqQualAndTags</a> ()</td></tr>
<tr class="memdesc:a9534f17fd0cfabf55455d733bb3715c2"><td class="mdescLeft">&#160;</td><td class="mdescRight">Remove the sequence, quality and alignment tags.  <a href="#a9534f17fd0cfabf55455d733bb3715c2">More...</a><br /></td></tr>
<tr class="separator:a9534f17fd0cfabf55455d733bb3715c2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a848ce46e52ffa3e2eba97345bf9edb22"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a848ce46e52ffa3e2eba97345bf9edb22"></a>
std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a848ce46e52ffa3e2eba97345bf9edb22">Sequence</a> () const </td></tr>
<tr class="memdesc:a848ce46e52ffa3e2eba97345bf9edb22"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the sequence of this read as a string. <br /></td></tr>
<tr class="separator:a848ce46e52ffa3e2eba97345bf9edb22"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3428aa6681ce6c279fe5379d1c68daf5"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a3428aa6681ce6c279fe5379d1c68daf5"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a3428aa6681ce6c279fe5379d1c68daf5">MeanPhred</a> () const </td></tr>
<tr class="memdesc:a3428aa6681ce6c279fe5379d1c68daf5"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the mean phred score. <br /></td></tr>
<tr class="separator:a3428aa6681ce6c279fe5379d1c68daf5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4ad19b4c9140e3e466a89cb3f1e346ce"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a4ad19b4c9140e3e466a89cb3f1e346ce"></a>
&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a4ad19b4c9140e3e466a89cb3f1e346ce">BamRecord</a> (const std::string &amp;name, const std::string &amp;seq, const std::string &amp;ref, const <a class="el" href="classSeqLib_1_1GenomicRegion.html">GenomicRegion</a> *gr)</td></tr>
<tr class="memdesc:a4ad19b4c9140e3e466a89cb3f1e346ce"><td class="mdescLeft">&#160;</td><td class="mdescRight">Do a smith waterman alignment. <br /></td></tr>
<tr class="separator:a4ad19b4c9140e3e466a89cb3f1e346ce"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a60509028b3c2b8adcb84d320f786b2ce"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a60509028b3c2b8adcb84d320f786b2ce"></a>
std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a60509028b3c2b8adcb84d320f786b2ce">Qualities</a> () const </td></tr>
<tr class="memdesc:a60509028b3c2b8adcb84d320f786b2ce"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the quality scores of this read as a string. <br /></td></tr>
<tr class="separator:a60509028b3c2b8adcb84d320f786b2ce"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a38bb030c2c0130e48c27bd33a3122bff"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a38bb030c2c0130e48c27bd33a3122bff"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a38bb030c2c0130e48c27bd33a3122bff">AlignmentPositionReverse</a> () const </td></tr>
<tr class="memdesc:a38bb030c2c0130e48c27bd33a3122bff"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the start of the alignment on the read, by removing soft-clips Do this in the reverse orientation though. <br /></td></tr>
<tr class="separator:a38bb030c2c0130e48c27bd33a3122bff"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a49fb5c59dbc95bb58f36ef119608c907"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a49fb5c59dbc95bb58f36ef119608c907"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a49fb5c59dbc95bb58f36ef119608c907">AlignmentEndPositionReverse</a> () const </td></tr>
<tr class="memdesc:a49fb5c59dbc95bb58f36ef119608c907"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the end of the alignment on the read, by removing soft-clips Do this in the reverse orientation though. <br /></td></tr>
<tr class="separator:a49fb5c59dbc95bb58f36ef119608c907"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a964f96c94ac8cdab7a74178cd6366dad"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a964f96c94ac8cdab7a74178cd6366dad"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a964f96c94ac8cdab7a74178cd6366dad">AlignmentPosition</a> () const </td></tr>
<tr class="memdesc:a964f96c94ac8cdab7a74178cd6366dad"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the start of the alignment on the read, by removing soft-clips. <br /></td></tr>
<tr class="separator:a964f96c94ac8cdab7a74178cd6366dad"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abc98a9ed00137658f5a6029627d2c691"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="abc98a9ed00137658f5a6029627d2c691"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#abc98a9ed00137658f5a6029627d2c691">AlignmentEndPosition</a> () const </td></tr>
<tr class="memdesc:abc98a9ed00137658f5a6029627d2c691"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the end of the alignment on the read, by removing soft-clips. <br /></td></tr>
<tr class="separator:abc98a9ed00137658f5a6029627d2c691"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a06eace5fde4de4946bafd0271dae4c7f"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a06eace5fde4de4946bafd0271dae4c7f"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a06eace5fde4de4946bafd0271dae4c7f">NumSoftClip</a> () const </td></tr>
<tr class="memdesc:a06eace5fde4de4946bafd0271dae4c7f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the number of soft clipped bases. <br /></td></tr>
<tr class="separator:a06eace5fde4de4946bafd0271dae4c7f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaa36a12f467323318b31011f9e72b8a6"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aaa36a12f467323318b31011f9e72b8a6"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#aaa36a12f467323318b31011f9e72b8a6">NumHardClip</a> () const </td></tr>
<tr class="memdesc:aaa36a12f467323318b31011f9e72b8a6"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the number of hard clipped bases. <br /></td></tr>
<tr class="separator:aaa36a12f467323318b31011f9e72b8a6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:affadd7f1657d8dd6838067294f9db2c9"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="affadd7f1657d8dd6838067294f9db2c9"></a>
int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#affadd7f1657d8dd6838067294f9db2c9">NumClip</a> () const </td></tr>
<tr class="memdesc:affadd7f1657d8dd6838067294f9db2c9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get the number of clipped bases (hard clipped and soft clipped) <br /></td></tr>
<tr class="separator:affadd7f1657d8dd6838067294f9db2c9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a35a81a15d80b25d069f315becdeac88f"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a35a81a15d80b25d069f315becdeac88f">GetZTag</a> (const std::string &amp;tag) const </td></tr>
<tr class="memdesc:a35a81a15d80b25d069f315becdeac88f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get a string (Z) tag.  <a href="#a35a81a15d80b25d069f315becdeac88f">More...</a><br /></td></tr>
<tr class="separator:a35a81a15d80b25d069f315becdeac88f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5d3fa999fe77fa623ed4ae37c3cc661f"><td class="memItemLeft" align="right" valign="top">std::vector&lt; int &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a5d3fa999fe77fa623ed4ae37c3cc661f">GetSmartIntTag</a> (const std::string &amp;tag) const </td></tr>
<tr class="memdesc:a5d3fa999fe77fa623ed4ae37c3cc661f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get a vector of type int from a Z tag delimited by "^".  <a href="#a5d3fa999fe77fa623ed4ae37c3cc661f">More...</a><br /></td></tr>
<tr class="separator:a5d3fa999fe77fa623ed4ae37c3cc661f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaab8d39d8ac409a3e51824af2e79e65a"><td class="memItemLeft" align="right" valign="top">std::vector&lt; double &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#aaab8d39d8ac409a3e51824af2e79e65a">GetSmartDoubleTag</a> (const std::string &amp;tag) const </td></tr>
<tr class="memdesc:aaab8d39d8ac409a3e51824af2e79e65a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get a vector of type double from a Z tag delimited by "x".  <a href="#aaab8d39d8ac409a3e51824af2e79e65a">More...</a><br /></td></tr>
<tr class="separator:aaab8d39d8ac409a3e51824af2e79e65a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab76ce21af871fe90c7b828aa8c3124b3"><td class="memItemLeft" align="right" valign="top">std::vector&lt; std::string &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ab76ce21af871fe90c7b828aa8c3124b3">GetSmartStringTag</a> (const std::string &amp;tag) const </td></tr>
<tr class="memdesc:ab76ce21af871fe90c7b828aa8c3124b3"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get a vector of strings from a Z tag delimited by "^".  <a href="#ab76ce21af871fe90c7b828aa8c3124b3">More...</a><br /></td></tr>
<tr class="separator:ab76ce21af871fe90c7b828aa8c3124b3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aabe3d89e97fccb2c1bdc23655ca396e4"><td class="memItemLeft" align="right" valign="top">int32_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#aabe3d89e97fccb2c1bdc23655ca396e4">GetIntTag</a> (const std::string &amp;tag) const </td></tr>
<tr class="memdesc:aabe3d89e97fccb2c1bdc23655ca396e4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Get an int (i) tag.  <a href="#aabe3d89e97fccb2c1bdc23655ca396e4">More...</a><br /></td></tr>
<tr class="separator:aabe3d89e97fccb2c1bdc23655ca396e4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac61f4df863735aa2ddfc7f54d66ba0f4"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ac61f4df863735aa2ddfc7f54d66ba0f4">AddZTag</a> (std::string tag, std::string val)</td></tr>
<tr class="memdesc:ac61f4df863735aa2ddfc7f54d66ba0f4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add a string (Z) tag.  <a href="#ac61f4df863735aa2ddfc7f54d66ba0f4">More...</a><br /></td></tr>
<tr class="separator:ac61f4df863735aa2ddfc7f54d66ba0f4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abe98e023ba9d4bf5420fc048772b2c81"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#abe98e023ba9d4bf5420fc048772b2c81">AddIntTag</a> (const std::string &amp;tag, int32_t val)</td></tr>
<tr class="memdesc:abe98e023ba9d4bf5420fc048772b2c81"><td class="mdescLeft">&#160;</td><td class="mdescRight">Add an int (i) tag.  <a href="#abe98e023ba9d4bf5420fc048772b2c81">More...</a><br /></td></tr>
<tr class="separator:abe98e023ba9d4bf5420fc048772b2c81"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a12e19312481cb669198668585ed263a0"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a12e19312481cb669198668585ed263a0">SetID</a> (int32_t id)</td></tr>
<tr class="memdesc:a12e19312481cb669198668585ed263a0"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the chr id number.  <a href="#a12e19312481cb669198668585ed263a0">More...</a><br /></td></tr>
<tr class="separator:a12e19312481cb669198668585ed263a0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1ef1594d552f155f92ba4fa6ffaf6c7d"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a1ef1594d552f155f92ba4fa6ffaf6c7d">SetPosition</a> (int32_t pos)</td></tr>
<tr class="memdesc:a1ef1594d552f155f92ba4fa6ffaf6c7d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Set the alignment start position.  <a href="#a1ef1594d552f155f92ba4fa6ffaf6c7d">More...</a><br /></td></tr>
<tr class="separator:a1ef1594d552f155f92ba4fa6ffaf6c7d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6c495a1389b8a22cd969a506ed366723"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a6c495a1389b8a22cd969a506ed366723"></a>
std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a6c495a1389b8a22cd969a506ed366723">CigarString</a> () const </td></tr>
<tr class="memdesc:a6c495a1389b8a22cd969a506ed366723"><td class="mdescLeft">&#160;</td><td class="mdescRight">Convert CIGAR to a string. <br /></td></tr>
<tr class="separator:a6c495a1389b8a22cd969a506ed366723"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1036efccc7eb4586738fea267085b386"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a1036efccc7eb4586738fea267085b386"></a>
std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a1036efccc7eb4586738fea267085b386">ChrName</a> () const </td></tr>
<tr class="memdesc:a1036efccc7eb4586738fea267085b386"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return a human readable chromosome name assuming chr is indexed from 0 (eg id 0 return "1") <br /></td></tr>
<tr class="separator:a1036efccc7eb4586738fea267085b386"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adccf293f5594d0a49aedffd71fe767e1"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#adccf293f5594d0a49aedffd71fe767e1">ChrName</a> (const <a class="el" href="classSeqLib_1_1BamHeader.html">SeqLib::BamHeader</a> &amp;h) const </td></tr>
<tr class="memdesc:adccf293f5594d0a49aedffd71fe767e1"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieve the human readable chromosome name.  <a href="#adccf293f5594d0a49aedffd71fe767e1">More...</a><br /></td></tr>
<tr class="separator:adccf293f5594d0a49aedffd71fe767e1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a39ffba1f5acfe961f71a4189b64fec4a"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a39ffba1f5acfe961f71a4189b64fec4a"></a>
std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a39ffba1f5acfe961f71a4189b64fec4a">Brief</a> () const </td></tr>
<tr class="memdesc:a39ffba1f5acfe961f71a4189b64fec4a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return a short description (chr:pos) of this read. <br /></td></tr>
<tr class="separator:a39ffba1f5acfe961f71a4189b64fec4a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1846b7cff76d1ab4920ca273e0bebb6f"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a1846b7cff76d1ab4920ca273e0bebb6f"></a>
std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a1846b7cff76d1ab4920ca273e0bebb6f">BriefMate</a> () const </td></tr>
<tr class="memdesc:a1846b7cff76d1ab4920ca273e0bebb6f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return a short description (chr:pos) of this read's mate. <br /></td></tr>
<tr class="separator:a1846b7cff76d1ab4920ca273e0bebb6f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae30193b6e1f7ab38e9a7379c62ee269c"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ae30193b6e1f7ab38e9a7379c62ee269c">RemoveTag</a> (const char *tag)</td></tr>
<tr class="memdesc:ae30193b6e1f7ab38e9a7379c62ee269c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Strip a particular alignment tag.  <a href="#ae30193b6e1f7ab38e9a7379c62ee269c">More...</a><br /></td></tr>
<tr class="separator:ae30193b6e1f7ab38e9a7379c62ee269c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a125b73c6ab23363d25daee2fd4eed65a"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a125b73c6ab23363d25daee2fd4eed65a"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a125b73c6ab23363d25daee2fd4eed65a">RemoveAllTags</a> ()</td></tr>
<tr class="memdesc:a125b73c6ab23363d25daee2fd4eed65a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Strip all of the alignment tags. <br /></td></tr>
<tr class="separator:a125b73c6ab23363d25daee2fd4eed65a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac8a8e932ebee9ad1abe2f4deafc23b31"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ac8a8e932ebee9ad1abe2f4deafc23b31"></a>
bam1_t *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#ac8a8e932ebee9ad1abe2f4deafc23b31">raw</a> () const </td></tr>
<tr class="memdesc:ac8a8e932ebee9ad1abe2f4deafc23b31"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the raw pointer. <br /></td></tr>
<tr class="separator:ac8a8e932ebee9ad1abe2f4deafc23b31"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acce3137c538a27c33962806312a8cdc3"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="acce3137c538a27c33962806312a8cdc3"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#acce3137c538a27c33962806312a8cdc3">OverlappingCoverage</a> (const <a class="el" href="classSeqLib_1_1BamRecord.html">BamRecord</a> &amp;r) const </td></tr>
<tr class="memdesc:acce3137c538a27c33962806312a8cdc3"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the number of bases on the reference that are covered by a match (M) on both reads. <br /></td></tr>
<tr class="separator:acce3137c538a27c33962806312a8cdc3"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="friends"></a>
Friends</h2></td></tr>
<tr class="memitem:aef0f85ab3addea99b38f0710c81e84d1"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="aef0f85ab3addea99b38f0710c81e84d1"></a>
class&#160;</td><td class="memItemRight" valign="bottom"><b>BLATWraper</b></td></tr>
<tr class="separator:aef0f85ab3addea99b38f0710c81e84d1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a631a77d9ad000faa2cc2a7d39b7d6c4e"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a631a77d9ad000faa2cc2a7d39b7d6c4e"></a>
class&#160;</td><td class="memItemRight" valign="bottom"><b>BWAWrapper</b></td></tr>
<tr class="separator:a631a77d9ad000faa2cc2a7d39b7d6c4e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a01e11933378b027ff89ef521e77079ba"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a01e11933378b027ff89ef521e77079ba"></a>
std::ostream &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classSeqLib_1_1BamRecord.html#a01e11933378b027ff89ef521e77079ba">operator&lt;&lt;</a> (std::ostream &amp;out, const <a class="el" href="classSeqLib_1_1BamRecord.html">BamRecord</a> &amp;r)</td></tr>
<tr class="memdesc:a01e11933378b027ff89ef521e77079ba"><td class="mdescLeft">&#160;</td><td class="mdescRight">Print a SAM-lite record for this alignment. <br /></td></tr>
<tr class="separator:a01e11933378b027ff89ef521e77079ba"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Class to store and interact with a SAM alignment record. </p>
<p>HTSLibrary reads are stored in the bam1_t struct. Memory allocation is taken care of by bam1_t init, and deallocation by destroy_bam1. This class is a C++ interface that automatically takes care of memory management for these C allocs/deallocs. The only member of <a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> is a bam1_t object. Alloc/dealloc is taken care of by the constructor and destructor. </p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a176dac15321681fbcc9f3b21bde3da16"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classSeqLib_1_1BamRecord.html">BamRecord</a> </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>name</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>seq</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="classSeqLib_1_1GenomicRegion.html">GenomicRegion</a> *&#160;</td>
          <td class="paramname"><em>gr</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="classSeqLib_1_1Cigar.html">Cigar</a> &amp;&#160;</td>
          <td class="paramname"><em>cig</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Construct a <a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> manually from a name, sequence, cigar and location. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">name</td><td>Name of the read </td></tr>
    <tr><td class="paramname">seq</td><td>Sequence of the read (compsed of ACTG or N). </td></tr>
    <tr><td class="paramname">gr</td><td>Location of the alignment </td></tr>
    <tr><td class="paramname">cig</td><td><a class="el" href="classSeqLib_1_1Cigar.html" title="CIGAR for a single gapped alignment. ">Cigar</a> alignment </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Throws</td><td>an invalid_argument exception if length of seq is not commensurate with number of query-bases consumed in cigar. </td></tr>
    <tr><td class="paramname">Throws</td><td>an invalid_argument exception if width of gr is not commensurate with number of reference-bases consumed in cigar. </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="abe98e023ba9d4bf5420fc048772b2c81"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void AddIntTag </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>tag</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int32_t&#160;</td>
          <td class="paramname"><em>val</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Add an int (i) tag. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">tag</td><td>Name of the tag. eg "XP" </td></tr>
    <tr><td class="paramname">val</td><td>Value for the tag </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ac61f4df863735aa2ddfc7f54d66ba0f4"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void AddZTag </td>
          <td>(</td>
          <td class="paramtype">std::string&#160;</td>
          <td class="paramname"><em>tag</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">std::string&#160;</td>
          <td class="paramname"><em>val</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Add a string (Z) tag. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">tag</td><td>Name of the tag. eg "XP" </td></tr>
    <tr><td class="paramname">val</td><td>Value for the tag </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="af1d3aa3591fce81d05b5515154181c70"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void assign </td>
          <td>(</td>
          <td class="paramtype">bam1_t *&#160;</td>
          <td class="paramname"><em>a</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Explicitly pass a bam1_t to the <a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a>. </p>
<p>The <a class="el" href="classSeqLib_1_1BamRecord.html" title="Class to store and interact with a SAM alignment record. ">BamRecord</a> now controls the memory, and will delete at destruction </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">a</td><td>An allocated bam1_t </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="adccf293f5594d0a49aedffd71fe767e1"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">std::string ChrName </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classSeqLib_1_1BamHeader.html">SeqLib::BamHeader</a> &amp;&#160;</td>
          <td class="paramname"><em>h</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Retrieve the human readable chromosome name. </p>
<p>Note that this requires that the header not be empty. If it is empty, assumes this ia chr1 based reference </p><dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Throws</td><td>an out_of_range exception if chr id is not in dictionary </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>Empty string if chr id &lt; 0, otherwise chromosome name from dictionary. </dd></dl>

<p>References <a class="el" href="classSeqLib_1_1BamHeader.html#afe1b3321a8153237a0512f6c87c4c20c">BamHeader::IDtoName()</a>, and <a class="el" href="classSeqLib_1_1BamHeader.html#a479432127ee77145cc19d6a2d1590821">BamHeader::isEmpty()</a>.</p>

</div>
</div>
<a class="anchor" id="a9534f17fd0cfabf55455d733bb3715c2"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void ClearSeqQualAndTags </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Remove the sequence, quality and alignment tags. </p>
<p>Make a more compact alignment stucture, without the string data </p>

</div>
</div>
<a class="anchor" id="a1d062cfbf0c0a2f1faa7238f0d5c013b"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int32_t CountBWAChimericAlignments </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Count the number of chimeric alignments by looking at XP and SA tags. </p>
<dl class="section note"><dt>Note</dt><dd>A secondary alignment is an alternative mapping. This may not work for non-BWA aligners that may not place the XP/SA tags. BWA-MEM used the XP tag prior to v0.7.5, and SA aftewards. </dd></dl>

</div>
</div>
<a class="anchor" id="ae056a71f4a71fe2d81218b1d93d1cfbe"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int32_t CountBWASecondaryAlignments </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Count the number of secondary alignments by looking at XA tag. </p>
<dl class="section note"><dt>Note</dt><dd>A secondary alignment is an alternative mapping. This may not work for non-BWA aligners that may not place the XA tag. </dd></dl>

</div>
</div>
<a class="anchor" id="aabe3d89e97fccb2c1bdc23655ca396e4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int32_t GetIntTag </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>tag</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Get an int (i) tag. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">tag</td><td>Name of the tag. eg "XP" </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>The value stored in the tag. Returns 0 if it does not exist. </dd></dl>

</div>
</div>
<a class="anchor" id="aaab8d39d8ac409a3e51824af2e79e65a"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt;double&gt; GetSmartDoubleTag </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>tag</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Get a vector of type double from a Z tag delimited by "x". </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">tag</td><td>Name of the tag eg "AL" </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>A vector of double elems, retrieved from the "^" delimited Z tag </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Throws</td><td>an invalid_argument if cannot convert delimited field val to double </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a5d3fa999fe77fa623ed4ae37c3cc661f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt;int&gt; GetSmartIntTag </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>tag</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Get a vector of type int from a Z tag delimited by "^". </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">tag</td><td>Name of the tag eg "AL" </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>A vector of ints, retrieved from the x delimited Z tag </dd></dl>
<dl class="exception"><dt>Exceptions</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Throws</td><td>an invalid_argument if cannot convert delimited field val to int </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ab76ce21af871fe90c7b828aa8c3124b3"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt;std::string&gt; GetSmartStringTag </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>tag</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Get a vector of strings from a Z tag delimited by "^". </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">tag</td><td>Name of the tag eg "CN" </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>A vector of strngs, retrieved from the x delimited Z tag </dd></dl>

</div>
</div>
<a class="anchor" id="a35a81a15d80b25d069f315becdeac88f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string GetZTag </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>tag</em></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Get a string (Z) tag. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">tag</td><td>Name of the tag. eg "XP" </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>The value stored in the tag. Returns empty string if it does not exist. </dd></dl>

<p>Referenced by <a class="el" href="classSeqLib_1_1BamRecord.html#acf667c1781f31e6518ed7e5dedb8c61a">BamRecord::ParseReadGroup()</a>.</p>

</div>
</div>
<a class="anchor" id="a663d6d827b5f26f89ee995afb814963a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int NumAlignedBases </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Return the number of "aligned bases" in the same style as BamTools. </p>
<p>BamTools reports AlignedBases, which for example returns the literal strings (for diff CIGARs): 3S5M - CTG 5M - CTAGC 3M1D3M - ATG-TGA 3M1I3M - ATGCTGA</p>
<dl class="section return"><dt>Returns</dt><dd>The number of M, D, X, = and I bases </dd></dl>

</div>
</div>
<a class="anchor" id="a7ce8b14d32f3167126f830b6c58904ed"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">int PairOrientation </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Get the relative pair orientations. </p>
<p>0 - FR (RFORIENTATION) (lower pos read is Fwd strand, higher is reverse) 1 - FF (FFORIENTATION) 2 - RF (RFORIENTATION) 3 - RR (RRORIENTATION) 4 - Undefined (UDORIENTATION) (unpaired or one/both is unmapped) </p>

<p>References <a class="el" href="classSeqLib_1_1BamRecord.html#a6a602cda286b646dd00170fec4bdd353">BamRecord::MatePosition()</a>, <a class="el" href="classSeqLib_1_1BamRecord.html#adcfaba5e50f7a799cb8d4d89fa5afae6">BamRecord::MateReverseFlag()</a>, <a class="el" href="classSeqLib_1_1BamRecord.html#ad4ab6b24a51712e0a8ce89e492d1d38e">BamRecord::PairMappedFlag()</a>, <a class="el" href="classSeqLib_1_1BamRecord.html#a6fd3f2218c3245356bd3f841616d208e">BamRecord::Position()</a>, and <a class="el" href="classSeqLib_1_1BamRecord.html#a20e39d0a44668c55b436cd7fd7d534d5">BamRecord::ReverseFlag()</a>.</p>

</div>
</div>
<a class="anchor" id="acf667c1781f31e6518ed7e5dedb8c61a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">std::string ParseReadGroup </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Get the read group, first from qname, then by RG tag. </p>
<dl class="section return"><dt>Returns</dt><dd>empty string if no readgroup found </dd></dl>

<p>References <a class="el" href="classSeqLib_1_1BamRecord.html#a35a81a15d80b25d069f315becdeac88f">BamRecord::GetZTag()</a>, and <a class="el" href="classSeqLib_1_1BamRecord.html#a98694bf297ed2b002773ededb9b7f0c9">BamRecord::Qname()</a>.</p>

</div>
</div>
<a class="anchor" id="ae09627143c815100d284c660962c5571"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string QualitySequence </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieve the quality trimmed seqeuence from QT tag if made. </p>
<p>Otherwise return normal seq </p>

</div>
</div>
<a class="anchor" id="ab479d342def7b30ecb7d15bc29c4cc12"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void QualityTrimmedSequence </td>
          <td>(</td>
          <td class="paramtype">int32_t&#160;</td>
          <td class="paramname"><em>qualTrim</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int32_t &amp;&#160;</td>
          <td class="paramname"><em>startpoint</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int32_t &amp;&#160;</td>
          <td class="paramname"><em>endpoint</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> const</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Trim the sequence down by removing bases from ends with low quality scores. </p>
<p>Stores the trimmed sequence in the GV tag, but does not affect any other part of read. </p><dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">qualTrim</td><td>Minimal quality score, zero-based (eg # == 2) </td></tr>
    <tr><td class="paramname">startpoint</td><td>Returns the new starting point for the sequence </td></tr>
    <tr><td class="paramname">endpoint</td><td>Return the new ending point for the sequence </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="ae30193b6e1f7ab38e9a7379c62ee269c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void RemoveTag </td>
          <td>(</td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>tag</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Strip a particular alignment tag. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">tag</td><td>Tag to remove </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a12e19312481cb669198668585ed263a0"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void SetID </td>
          <td>(</td>
          <td class="paramtype">int32_t&#160;</td>
          <td class="paramname"><em>id</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Set the chr id number. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">id</td><td>Chromosome id. Typically is 0 for chr1, etc </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a class="anchor" id="a1ef1594d552f155f92ba4fa6ffaf6c7d"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void SetPosition </td>
          <td>(</td>
          <td class="paramtype">int32_t&#160;</td>
          <td class="paramname"><em>pos</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Set the alignment start position. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">pos</td><td>Alignment start position </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>SeqLib/<a class="el" href="BamRecord_8h_source.html">BamRecord.h</a></li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sat Oct 1 2016 19:04:34 for SeqLib by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.8
</small></address>
</body>
</html>
